--- a/Lib/os.py
+++ b/Lib/os.py
@@ -602,6 +602,7 @@
         fullname = path.join(dir, file)
         try:
             exec_func(fullname, *argrest)
+            return
         except (FileNotFoundError, NotADirectoryError) as e:
             last_exc = e
         except OSError as e:
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -781,7 +781,7 @@
 
     """ Interface to the system's uname command.
     """
-    if sys.platform in ('dos', 'win32', 'win16'):
+    if sys.platform in ('dos', 'win32', 'win16', 'darwin'):
         # XXX Others too ?
         return default
     try:
@@ -804,7 +804,7 @@
         default in case the command should fail.
 
     """
-    if sys.platform in ('dos', 'win32', 'win16'):
+    if sys.platform in ('dos', 'win32', 'win16', 'darwin'):
         # XXX Others too ?
         return default
     target = _follow_symlinks(target)
--- a/Lib/subprocess.py
+++ b/Lib/subprocess.py
@@ -851,6 +851,7 @@
         if not self._child_created:
             # We didn't get to successfully create a child process.
             return
+        return
         if self.returncode is None:
             # Not reading subprocess exit status creates a zombie process which
             # is only destroyed at the parent python process exit
@@ -1394,6 +1395,7 @@
             else:
                 args = list(args)
 
+            shell = False
             if shell:
                 # On Android the default shell is at '/system/bin/sh'.
                 unix_shell = ('/system/bin/sh' if
@@ -1452,6 +1454,7 @@
                             errpipe_read, errpipe_write,
                             restore_signals, start_new_session, preexec_fn)
                     self._child_created = True
+                    return
                 finally:
                     # be sure the FD is closed no matter what
                     os.close(errpipe_write)
@@ -1480,6 +1483,7 @@
             finally:
                 # be sure the FD is closed no matter what
                 os.close(errpipe_read)
+                return
 
             if errpipe_data:
                 try:
--- a/Lib/urllib/request.py
+++ b/Lib/urllib/request.py
@@ -2580,11 +2580,9 @@
 
 
 if sys.platform == 'darwin':
-    from _scproxy import _get_proxy_settings, _get_proxies
 
     def proxy_bypass_macosx_sysconf(host):
-        proxy_settings = _get_proxy_settings()
-        return _proxy_bypass_macosx_sysconf(host, proxy_settings)
+        return False
 
     def getproxies_macosx_sysconf():
         """Return a dictionary of scheme -> proxy server URL mappings.
@@ -2592,7 +2590,7 @@
         This function uses the MacOSX framework SystemConfiguration
         to fetch the proxy information.
         """
-        return _get_proxies()
+        return {}
 
 
 
@@ -2605,9 +2603,9 @@
         """
         proxies = getproxies_environment()
         if proxies:
-            return proxy_bypass_environment(host, proxies)
+            return False
         else:
-            return proxy_bypass_macosx_sysconf(host)
+            return False
 
     def getproxies():
         return getproxies_environment() or getproxies_macosx_sysconf()
--- a/Modules/_posixsubprocess.c
+++ b/Modules/_posixsubprocess.c
@@ -49,6 +49,9 @@
 
 #define POSIX_CALL(call)   do { if ((call) == -1) goto error; } while (0)
 
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#endif
 
 /* If gc was disabled, call gc.enable().  Return 0 on success. */
 static int
@@ -419,14 +422,14 @@
         goto error;
 
     /* Close parent's pipe ends. */
-    if (p2cwrite != -1)
+/*    if (p2cwrite != -1)
         POSIX_CALL(close(p2cwrite));
     if (c2pread != -1)
         POSIX_CALL(close(c2pread));
     if (errread != -1)
         POSIX_CALL(close(errread));
     POSIX_CALL(close(errpipe_read));
-
+*/
     /* When duping fds, if there arises a situation where one of the fds is
        either 0, 1 or 2, it is possible that it is overwritten (#12607). */
     if (c2pwrite == 0) {
@@ -471,6 +474,8 @@
     /* We no longer manually close p2cread, c2pwrite, and errwrite here as
      * _close_open_fds takes care when it is not already non-inheritable. */
 
+    const char *currentDir = getcwd(NULL, 0);
+
     if (cwd)
         POSIX_CALL(chdir(cwd));
 
@@ -496,13 +501,13 @@
             errno = 0;  /* We don't want to report an OSError. */
             goto error;
         }
-        /* Py_DECREF(result); - We're about to exec so why bother? */
+        Py_DECREF(result); /* - We're about to exec so why bother? */
     }
 
     /* close FDs after executing preexec_fn, which might open FDs */
     if (close_fds) {
         /* TODO HP-UX could use pstat_getproc() if anyone cares about it. */
-        _close_open_fds(3, py_fds_to_keep);
+//        _close_open_fds(3, py_fds_to_keep);
     }
 
     /* This loop matches the Lib/os.py _execvpe()'s PATH search when */
@@ -511,20 +516,34 @@
     for (i = 0; exec_array[i] != NULL; ++i) {
         const char *executable = exec_array[i];
         if (envp) {
+#if defined(TARGET_OS_TV)
+            saved_errno = ENOTSUP;
+#elif defined(TARGET_OS_IOS)
             execve(executable, argv, envp);
+            saved_errno = execve(executable, argv, envp);
+#endif
+            break;
         } else {
+#if defined(TARGET_OS_TV)
+            saved_errno = ENOTSUP;
+#elif defined(TARGET_OS_IOS)
             execv(executable, argv);
+            saved_errno = execv(executable, argv);
+#endif
+            break;
         }
         if (errno != ENOENT && errno != ENOTDIR && saved_errno == 0) {
             saved_errno = errno;
         }
     }
+    if (saved_errno == 0) return;
     /* Report the first exec error, not the last. */
     if (saved_errno)
         errno = saved_errno;
 
 error:
     saved_errno = errno;
+    chdir(currentDir);
     /* Report the posix error to our parent process. */
     /* We ignore all write() return values as the total size of our writes is
        less than PIPEBUF and we cannot do anything about an error anyways.
@@ -683,7 +702,8 @@
         need_after_fork = 1;
     }
 
-    pid = fork();
+    pid = -1;
+    errno = ENOTSUP;
     if (pid == 0) {
         /* Child process */
         /*
--- a/Modules/getpath.c
+++ b/Modules/getpath.c
@@ -593,8 +593,10 @@
       * will fail if a relative path was used. but in that case,
       * absolutize() should help us out below
       */
-    else if(0 == _NSGetExecutablePath(execpath, &nsexeclength) &&
-            execpath[0] == SEP)
+    char *iospath = Py_GETENV("PYTHONHOME");
+    char *prog = Py_GETENV("PYTHONEXECUTABLE");
+    sprintf(execpath, "%s/bin/%s", iospath, prog);
+    if (execpath[0] == SEP)
     {
         size_t len;
         wchar_t *path = Py_DecodeLocale(execpath, &len);
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -197,6 +197,19 @@
 #endif  /* _MSC_VER */
 #endif  /* ! __WATCOMC__ || __QNX__ */
 
+#  undef HAVE_EXECV
+#  undef HAVE_FORK
+#  undef HAVE_FORK1
+#  undef HAVE_FORKPTY
+#  undef HAVE_GETGROUPS
+#  undef HAVE_SCHED_H
+#  undef HAVE_SENDFILE
+#  undef HAVE_SETPRIORITY
+#  undef HAVE_SPAWNV
+#  undef HAVE_SYSTEM
+#  undef HAVE_WAIT
+#  undef HAVE_WAIT3
+#  undef HAVE_WAIT4
 
 /*[clinic input]
 # one of the few times we lie about this name!
@@ -475,6 +488,10 @@
 }
 #endif
 
+void PyOS_BeforeFork(void){}
+void PyOS_AfterFork_Parent(void){}
+void PyOS_AfterFork_Child(void){}
+
 /* Legacy wrapper */
 void
 PyOS_AfterFork(void)
@@ -5040,6 +5057,8 @@
 #endif
     _Py_END_SUPPRESS_IPH
 
+    Py_RETURN_NONE;
+
     /* If we get here it's definitely an error */
 
     free_string_array(argvlist, argc);
@@ -5116,6 +5135,12 @@
 #endif
     _Py_END_SUPPRESS_IPH
 
+    while (--envc >= 0)
+        PyMem_DEL(envlist[envc]);
+    PyMem_DEL(envlist);
+    // iOS: we return now
+    Py_RETURN_NONE;
+
     /* If we get here it's definitely an error */
 
     posix_path_error(path);
@@ -5416,7 +5441,7 @@
     pid_t pid;
 
     PyOS_BeforeFork();
-    pid = fork1();
+    pid = -1;
     if (pid == 0) {
         /* child: this clobbers and resets the import lock. */
         PyOS_AfterFork_Child();
@@ -5447,7 +5472,7 @@
     pid_t pid;
 
     PyOS_BeforeFork();
-    pid = fork();
+    pid = -1;
     if (pid == 0) {
         /* child: this clobbers and resets the import lock. */
         PyOS_AfterFork_Child();
