--- a/Modules/_ctypes/cfield.c
+++ b/Modules/_ctypes/cfield.c
@@ -1634,6 +1634,7 @@
 */
 
 /* align and size are bogus for void, but they must not be zero */
+/*
 ffi_type ffi_type_void = { 1, 1, FFI_TYPE_VOID };
 
 ffi_type ffi_type_uint8 = { 1, 1, FFI_TYPE_UINT8 };
@@ -1650,14 +1651,15 @@
 
 ffi_type ffi_type_float = { sizeof(float), FLOAT_ALIGN, FFI_TYPE_FLOAT };
 ffi_type ffi_type_double = { sizeof(double), DOUBLE_ALIGN, FFI_TYPE_DOUBLE };
+*/
 
 #ifdef ffi_type_longdouble
 #undef ffi_type_longdouble
 #endif
-  /* This is already defined on OSX */
+  /* This is already defined on OSX
 ffi_type ffi_type_longdouble = { sizeof(long double), LONGDOUBLE_ALIGN,
                                  FFI_TYPE_LONGDOUBLE };
 
 ffi_type ffi_type_pointer = { sizeof(void *), VOID_P_ALIGN, FFI_TYPE_POINTER };
-
+  */
 /*---------------- EOF ----------------*/
--- a/Modules/_ctypes/malloc_closure.c
+++ b/Modules/_ctypes/malloc_closure.c
@@ -89,6 +89,7 @@
 /******************************************************************/
 
 /* put the item back into the free list */
+/*
 void ffi_closure_free(void *p)
 {
     ITEM *item = (ITEM *)p;
@@ -96,7 +97,6 @@
     free_list = item;
 }
 
-/* return one item from the free list, allocating more if needed */
 void *ffi_closure_alloc(size_t ignored, void** codeloc)
 {
     ITEM *item;
@@ -109,3 +109,4 @@
     *codeloc = (void *)item;
     return (void *)item;
 }
+*/
