--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -10365,7 +10365,7 @@
 os_abort_impl(PyObject *module)
 /*[clinic end generated code: output=dcf52586dad2467c input=cf2c7d98bc504047]*/
 {
-    abort();
+    // abort();
     /*NOTREACHED*/
     Py_FatalError("abort() called from Python code didn't abort!");
     return NULL;
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -1697,7 +1697,7 @@
     DebugBreak();
 #endif
 #endif /* MS_WINDOWS */
-    abort();
+	// abort();
 }

 /* Clean up and exit */
