--- a/include/my_global.h	2015-09-19 16:09:00.000000000 -0400
+++ b/include/my_global.h-org	2015-09-19 16:05:28.000000000 -0400
@@ -135,7 +135,7 @@
 #  if defined(__i386__) || defined(__ppc__) || defined(__arm__)
 #    define SIZEOF_CHARP 4
 #    define SIZEOF_LONG 4
-#  elif defined(__x86_64__) || defined(__ppc64__)
+#  elif defined(__x86_64__) || defined(__ppc64__) || defined(__aarch64__)
 #    define SIZEOF_CHARP 8
 #    define SIZEOF_LONG 8
 #  else
