--- include/nfsc/libnfs.h.orig	2013-05-28 21:59:32.000000000 +0200
+++ include/nfsc/libnfs.h	2013-05-28 21:59:56.000000000 +0200
@@ -18,12 +18,7 @@
  * This is the highlevel interface to access NFS resources using a posix-like interface
  */
 #include <stdint.h>
-#if defined(ANDROID)
 #include <sys/time.h>
-#endif
-#if defined(AROS)
-#include <sys/time.h>
-#endif
 
 struct nfs_context;
 struct rpc_context;
