--- afpfs-ng-0.8.1/lib/lowlevel.c	2011-10-01 20:36:55.000000000 +0200
+++ afpfs-ng-0.8.1/lib/lowlevel.c	2011-10-01 20:46:00.000000000 +0200
@@ -579,7 +579,7 @@
 		return -EIO;
 	}
 
-	if (volume->server->using_version->av_number>=30)
+	if (volume->server->using_version->av_number>=30 && fp.unixprivs.permissions != 0)
 		stbuf->st_mode |= fp.unixprivs.permissions;
 	else
 		set_nonunix_perms(&stbuf->st_mode,&fp);
