diff -Naur Imaging-1.1.7/setup.py Imaging-1.1.7.patch/setup.py
--- Imaging-1.1.7/setup.py	2009-11-15 17:06:10.000000000 +0100
+++ Imaging-1.1.7.patch/setup.py	2011-04-17 09:31:07.545708559 +0200
@@ -83,7 +83,7 @@
 # --------------------------------------------------------------------
 
 from distutils import sysconfig
-from distutils.core import Extension, setup
+from setuptools import Extension, setup
 from distutils.command.build_ext import build_ext
 
 try:
