diff -Naur Imaging-1.1.7/setup.py Imaging-1.1.7.patch/setup.py
--- Imaging-1.1.7/setup.py	2011-04-17 09:31:07.000000000 +0200
+++ Imaging-1.1.7.patch/setup.py	2011-04-17 09:54:20.880047886 +0200
@@ -147,7 +147,6 @@
             add_directory(library_dirs, "/opt/local/lib")
             add_directory(include_dirs, "/opt/local/include")
 
-        add_directory(library_dirs, "/usr/local/lib")
         # FIXME: check /opt/stuff directories here?
 
         prefix = sysconfig.get_config_var("prefix")
@@ -199,22 +198,6 @@
             add_directory(include_dirs, include_root)
 
         #
-        # add standard directories
-
-        # look for tcl specific subdirectory (e.g debian)
-        if _tkinter:
-            tcl_dir = "/usr/include/tcl" + TCL_VERSION
-            if os.path.isfile(os.path.join(tcl_dir, "tk.h")):
-                add_directory(include_dirs, tcl_dir)
-
-        # standard locations
-        add_directory(library_dirs, "/usr/local/lib")
-        add_directory(include_dirs, "/usr/local/include")
-
-        add_directory(library_dirs, "/usr/lib")
-        add_directory(include_dirs, "/usr/include")
-
-        #
         # insert new dirs *before* default libs, to avoid conflicts
         # between Python PYD stub libs and real libraries
 
