diff -rupN Imaging-1.1.7/libImaging/Access.c Imaging-1.1.7-mod/libImaging/Access.c
--- Imaging-1.1.7/libImaging/Access.c	2009-10-31 20:44:12.000000000 -0400
+++ Imaging-1.1.7-mod/libImaging/Access.c	2011-10-10 19:40:22.000000000 -0400
@@ -32,7 +32,7 @@ add_item(const char* mode)
 {
     UINT32 i = hash(mode);
     /* printf("hash %s => %d\n", mode, i); */
-    if (access_table[i].mode) {
+    if (access_table[i].mode && (strcmp(mode, access_table[i].mode) != 0)) {
         fprintf(stderr, "AccessInit: hash collision: %d for both %s and %s\n",
                 i, mode, access_table[i].mode);
         exit(1);

