--- src/util/daemonizeorig.cpp	2014-11-27 18:36:30.474678137 +0000
+++ src/util/daemonize.cpp	2014-11-27 18:36:44.226678082 +0000
@@ -35,10 +35,12 @@
   if (setsid() < 0)
     fprintf(stderr, "setsid(): %s", GetErrno().c_str());
 
+#if !defined(__ANDROID__)
   //route stdout and stderr to /dev/null
   fclose(stdout);
   stdout = fopen("/dev/null", "w");
   fclose(stderr);
   stderr = fopen("/dev/null", "w");
+#endif
 }
 
--- src/lib/boblightorig.h	2014-11-27 18:38:37.174677063 +0000
+++ src/lib/boblight.h	2014-11-27 18:40:14.810676123 +0000
@@ -61,7 +61,7 @@
     //gets a functionpointer from dlsym, and returns char* from dlerror if it didn't work
     #define BOBLIGHT_FUNCTION(returnvalue, name, arguments) \
     name = BOBLIGHT_CAST(returnvalue (*) arguments)(dlsym(p_boblight, #name)); \
-                        { char* error = dlerror(); if (error) return error; }
+                        { char* error = (char *)dlerror(); if (error) return error; }
 
     void* p_boblight = NULL; //where we put the lib
 
@@ -79,7 +79,7 @@
 
       p_boblight = dlopen(filename, RTLD_NOW);
       if (p_boblight == NULL)
-        return dlerror();
+        return (char *)dlerror();
 
       //generate dlsym lines
       #include "boblight-functions.h"
