--- a/src/device/deviceltbl.cpp
+++ b/src/device/deviceltbl.cpp
@@ -51,7 +51,7 @@
 
 bool CDeviceLtbl::WriteOutput()
 {
-  uint8_t prefix[4] = {0x55, 0xAA, 0x00, m_channels.size()};
+  uint8_t prefix[4] = {0x55, 0xAA, 0x00, (uint8_t)m_channels.size()};
 
   //get the channel values from the clienshandler
   int64_t now = GetTimeUs();
@@ -114,7 +114,7 @@
   uint8_t buff[512];
   uint8_t prefix[2]    = {0x55, 0xAA};
   uint8_t open[2]      = {0x83, 0x00};
-  uint8_t getvalues[4] = {0x81, 0x02, 0x00, m_channels.size()};
+  uint8_t getvalues[4] = {0x81, 0x02, 0x00, (uint8_t)m_channels.size()};
 
   if (m_isopened)
     return true; //nothing to do here
--- a/src/util/daemonize.cpp
+++ b/src/util/daemonize.cpp
@@ -25,7 +25,7 @@
 void Daemonize()
 {
   //fork a child process
-  pid_t pid = fork();
+  pid_t pid = -1;//fork();
   if (pid == -1)
     fprintf(stderr, "fork(): %s", GetErrno().c_str());
   else if (pid > 0)
--- a/src/util/tcpsocket.cpp
+++ b/src/util/tcpsocket.cpp
@@ -31,8 +31,6 @@
 #include "tcpsocket.h"
 #include "misc.h"
 
-using namespace std;
-
 void CTcpData::SetData(uint8_t* data, int size, bool append)
 {
   CopyData(reinterpret_cast<char*>(data), size, append);
