--- a/lib/ldb/wscript
+++ b/lib/ldb/wscript
@@ -381,10 +381,6 @@
                      install=False)

     bld.SAMBA_BINARY('ldb_key_value_sub_txn_tdb_test',
-                     bld.SUBDIR('ldb_key_value',
-                         '''ldb_kv_search.c
-                            ldb_kv_index.c
-                            ldb_kv_cache.c''') +
                      'tests/ldb_key_value_sub_txn_test.c',
                      cflags='-DTEST_BE=\"tdb\"',
                      deps='cmocka ldb ldb_tdb_err_map',
