--- a/pidl/lib/Parse/Pidl/ODL.pm
+++ b/pidl/lib/Parse/Pidl/ODL.pm
@@ -70,7 +70,7 @@
 					next;
 				}
 				my $podl = Parse::Pidl::IDL::parse_file($idl_path, $opt_incdirs);
-				if (defined(@$podl)) {
+				if (@$podl) {
 					require Parse::Pidl::Typelist;
 					my $basename = basename($idl_path, ".idl");
 
--- a/pidl/pidl
+++ b/pidl/pidl
@@ -604,7 +604,7 @@
 		require Parse::Pidl::IDL;
 
 		$pidl = Parse::Pidl::IDL::parse_file($idl_file, \@opt_incdirs);
-		defined @$pidl || die "Failed to parse $idl_file";
+		@$pidl || die "Failed to parse $idl_file";
 	}
 
 	require Parse::Pidl::Typelist;
