--- setup_o.py	2017-01-18 17:54:03.361747272 +0100
+++ setup.py	2017-01-18 17:54:33.337935416 +0100
@@ -20,10 +20,10 @@
 # SOFTWARE.
 # ===================================================================
 
-try:
-    from setuptools import Extension, Command, setup
-except ImportError:
-    from distutils.core import Extension, Command, setup
+#try:
+#    from setuptools import Extension, Command, setup
+#except ImportError:
+from distutils.core import Extension, Command, setup
 from distutils.command.build_ext import build_ext
 from distutils.command.build import build
 from distutils.command.install_lib import install_lib
@@ -464,16 +464,16 @@
         "Crypto.Signature",
         "Crypto.Util",
         "Crypto.Math",
-        "Crypto.SelfTest",
-        "Crypto.SelfTest.Cipher",
-        "Crypto.SelfTest.Hash",
-        "Crypto.SelfTest.IO",
-        "Crypto.SelfTest.Protocol",
-        "Crypto.SelfTest.PublicKey",
-        "Crypto.SelfTest.Random",
-        "Crypto.SelfTest.Signature",
-        "Crypto.SelfTest.Util",
-        "Crypto.SelfTest.Math",
+#        "Crypto.SelfTest",
+#        "Crypto.SelfTest.Cipher",
+#        "Crypto.SelfTest.Hash",
+#        "Crypto.SelfTest.IO",
+#        "Crypto.SelfTest.Protocol",
+#        "Crypto.SelfTest.PublicKey",
+#        "Crypto.SelfTest.Random",
+#        "Crypto.SelfTest.Signature",
+#        "Crypto.SelfTest.Util",
+#        "Crypto.SelfTest.Math",
         ],
     package_dir = { "Crypto": "lib/Crypto" },
     package_data = {
