--- a/setup.py
+++ b/setup.py
@@ -22,14 +22,14 @@
 
 from __future__ import print_function
 
-try:
-    from setuptools import Extension, Command, setup
-    from setuptools.command.build_ext import build_ext
-    from setuptools.command.build_py import build_py
-except ImportError:
-    from distutils.core import Extension, Command, setup
-    from distutils.command.build_ext import build_ext
-    from distutils.command.build_py import build_py
+#try:
+#    from setuptools import Extension, Command, setup
+#    from setuptools.command.build_ext import build_ext
+#    from setuptools.command.build_py import build_py
+#except ImportError:
+from distutils.core import Extension, Command, setup
+from distutils.command.build_ext import build_ext
+from distutils.command.build_py import build_py
 
 import re
 import os
@@ -252,16 +252,16 @@
     "Crypto.Signature",
     "Crypto.Util",
     "Crypto.Math",
-    "Crypto.SelfTest",
-    "Crypto.SelfTest.Cipher",
-    "Crypto.SelfTest.Hash",
-    "Crypto.SelfTest.IO",
-    "Crypto.SelfTest.Protocol",
-    "Crypto.SelfTest.PublicKey",
-    "Crypto.SelfTest.Random",
-    "Crypto.SelfTest.Signature",
-    "Crypto.SelfTest.Util",
-    "Crypto.SelfTest.Math",
+#    "Crypto.SelfTest",
+#    "Crypto.SelfTest.Cipher",
+#    "Crypto.SelfTest.Hash",
+#    "Crypto.SelfTest.IO",
+#    "Crypto.SelfTest.Protocol",
+#    "Crypto.SelfTest.PublicKey",
+#    "Crypto.SelfTest.Random",
+#    "Crypto.SelfTest.Signature",
+#    "Crypto.SelfTest.Util",
+#    "Crypto.SelfTest.Math",
 ]
 package_data = {
     "Crypto" : [ "py.typed", "*.pyi" ],
