--- a/lib/Crypto/Util/_raw_api.py
+++ b/lib/Crypto/Util/_raw_api.py
@@ -170,11 +170,6 @@
     null_pointer = None
     cached_architecture = []
 
-    def c_ubyte(c):
-        if not (0 <= c < 256):
-            raise OverflowError()
-        return ctypes.c_ubyte(c)
-
     def load_lib(name, cdecl):
         if not cached_architecture:
             # platform.architecture() creates a subprocess, so caching the
@@ -200,8 +195,8 @@
     _c_ssize_t = ctypes.c_ssize_t
 
     _PyBUF_SIMPLE = 0
-    _PyObject_GetBuffer = ctypes.pythonapi.PyObject_GetBuffer
-    _PyBuffer_Release = ctypes.pythonapi.PyBuffer_Release
+    #_PyObject_GetBuffer = ctypes.pythonapi.PyObject_GetBuffer
+    #_PyBuffer_Release = ctypes.pythonapi.PyBuffer_Release
     _py_object = ctypes.py_object
     _c_ssize_p = ctypes.POINTER(_c_ssize_t)
 
@@ -229,15 +224,9 @@
     def c_uint8_ptr(data):
         if byte_string(data) or isinstance(data, _Array):
             return data
-        elif isinstance(data, _buffer_type):
-            obj = _py_object(data)
-            buf = _Py_buffer()
-            _PyObject_GetBuffer(obj, byref(buf), _PyBUF_SIMPLE)
-            try:
-                buffer_type = ctypes.c_ubyte * buf.len
-                return buffer_type.from_address(buf.buf)
-            finally:
-                _PyBuffer_Release(byref(buf))
+        elif isinstance(data, bytearray):
+            local_type = c_ubyte * len(data)
+            return local_type.from_buffer(data)
         else:
             raise TypeError("Object type %s cannot be passed to C code" % type(data))
 
