--- a/lib/Crypto/Util/_file_system.py
+++ b/lib/Crypto/Util/_file_system.py
@@ -28,7 +28,7 @@
 # POSSIBILITY OF SUCH DAMAGE.
 # ===================================================================
 
-import os
+import os, sys
 
 
 def pycryptodome_filename(dir_comps, filename):
@@ -45,10 +45,8 @@ def pycryptodome_filename(dir_comps, filename):
     if dir_comps[0] != "Cryptodome":
         raise ValueError("Only available for modules under 'Cryptodome'")
 
-    dir_comps = list(dir_comps[1:]) + [filename]
-
-    util_lib, _ = os.path.split(os.path.abspath(__file__))
-    root_lib = os.path.join(util_lib, "..")
-
-    return os.path.join(root_lib, *dir_comps)
+    appBundlePath = os.path.dirname(sys.executable)
+    libStem, _ = os.path.splitext(filename)
+    pyFramework = f"{'.'.join(dir_comps)}.{libStem}"
+    return os.path.join(appBundlePath, "Frameworks", f"{pyFramework}.framework", pyFramework)
 
