--- a/setup.py
+++ b/setup.py
@@ -266,6 +266,16 @@
         # Disable any assembly in libtomcrypt files
         self.compiler.define_macro("LTC_NO_ASM")
 
+        compiler = os.environ.get('CC')
+        args = {}
+        # unfortunately, distutils doesn't let us provide separate C and C++
+        # compilers
+        if compiler is not None:
+            ccshared = os.environ.get('CCSHARED') or ''
+            cflags = os.environ.get('CFLAGS') or ''
+            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
+        self.compiler.set_executables(**args)
+
         # Detect which modules should be compiled
         self.detect_modules()
 
