diff -ruN armeabi-v7a/configure.ac armeabi-v7a-good/configure.ac
--- armeabi-v7a/configure.ac	2011-05-20 01:44:18.000000000 -0400
+++ armeabi-v7a-good/configure.ac	2012-06-19 01:58:45.483524179 -0400
@@ -5,7 +5,7 @@
 AC_PREREQ([2.59])
 
 AC_CONFIG_HEADERS([config.h])
-CFLAGS="${CFLAGS} -Wall -Werror"
+CFLAGS="${CFLAGS} -Wall"
 
 ################################################################################
 # Set the shared versioning info, according to section 6.3 of the libtool info #
@@ -149,8 +149,6 @@
 # Check for libraries                                                          #
 ################################################################################
 
-AC_CHECK_LIB([pthread], [pthread_create])
-
 # Check for libdl
 AC_SEARCH_LIBS([dlopen], [dl], [], [AC_MSG_ERROR([libdl is required])])
 
diff -ruN armeabi-v7a/src/Makefile.am armeabi-v7a-good/src/Makefile.am
--- armeabi-v7a/src/Makefile.am	2011-05-20 01:28:17.000000000 -0400
+++ armeabi-v7a-good/src/Makefile.am	2012-06-19 01:59:23.603524155 -0400
@@ -27,7 +27,7 @@
 
 libomxil_bellagio_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/base -I$(srcdir)/core_extensions \
                               -DINSTALL_PATH_STR=\"$(plugindir)\" -DOMX_LOADERS_DIRNAME=\"$(libdir)/omxloaders\/\"
-libomxil_bellagio_la_LIBADD = base/libomxbase.la core_extensions/libomxcoreext.la -lpthread
+libomxil_bellagio_la_LIBADD = base/libomxbase.la core_extensions/libomxcoreext.la
 libomxil_bellagio_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
 
 include_extradir = $(includedir)/bellagio
