From eaa6256c161f1f6d48b222b171f68869dace8f3f Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Fri, 9 Jul 2010 16:49:01 -0400
Subject: [PATCH 12/24] include stdint.h

---
 libavutil/common.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/libavutil/common.h b/libavutil/common.h
index 84290c6..5b73579 100644
--- a/libavutil/common.h
+++ b/libavutil/common.h
@@ -29,6 +29,7 @@
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
+#include <stdint.h>
 #include <limits.h>
 #include <math.h>
 #include <stdio.h>
-- 
1.7.9.4

