From 9066d83f0f237d90fe0c2d9b6202bb8cd59a6562 Mon Sep 17 00:00:00 2001
From: Allan Kristensen <allank@tigerdyr.com>
Date: Fri, 21 Dec 2012 21:04:18 +0100
Subject: [PATCH 1/1] [ffmpeg] fixed missing S_DVBSUB subtitles

---
 lib/ffmpeg/libavformat/matroska.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/ffmpeg/libavformat/matroska.c b/lib/ffmpeg/libavformat/matroska.c
index 2f5b617..11c2fa5 100644
--- a/lib/ffmpeg/libavformat/matroska.c
+++ b/lib/ffmpeg/libavformat/matroska.c
@@ -59,6 +59,7 @@ const CodecTags ff_mkv_codec_tags[]={
     {"S_ASS"            , CODEC_ID_SSA},
     {"S_SSA"            , CODEC_ID_SSA},
     {"S_VOBSUB"         , CODEC_ID_DVD_SUBTITLE},
+    {"S_DVBSUB"         , CODEC_ID_DVB_SUBTITLE},
     {"S_HDMV/PGS"       , CODEC_ID_HDMV_PGS_SUBTITLE},
 
     {"V_DIRAC"          , CODEC_ID_DIRAC},
-- 
1.8.0

