--- a/m4/dpkg-progs.m4
+++ b/m4/dpkg-progs.m4
@@ -7,7 +7,7 @@
 # Locate perl interpreter in the path
 AC_DEFUN([DPKG_PROG_PERL], [
   AC_ARG_VAR([PERL], [Perl interpreter])dnl
-  m4_define([_PERL_MIN_VERSION], [5.20.2])
+  m4_define([_PERL_MIN_VERSION], [5.18.2])
   AC_SUBST([PERL_MIN_VERSION], [_PERL_MIN_VERSION])
   AC_CACHE_CHECK([for perl >= _PERL_MIN_VERSION], [ac_cv_path_PERL], [
     AC_PATH_PROGS_FEATURE_CHECK([PERL], [perl], [
