--- src/misc.c	2011-05-31 10:29:52.000000000 -0400
+++ src/misc.c	2013-01-03 00:37:37.652737345 -0500
@@ -207,6 +208,9 @@
 
 char *ssh_get_user_home_dir(void) {
   char *szPath = NULL;
+#ifdef ANDROID
+  return strdup(getenv("HOME"));
+#else
   struct passwd pwd;
   struct passwd *pwdbuf;
   char buf[NSS_BUFLEN_PASSWD];
@@ -225,7 +222,7 @@
   }
 
   szPath = strdup(pwd.pw_dir);
-
+#endif
   return szPath;
 }
 
@@ -239,6 +236,17 @@
 }
 
 char *ssh_get_local_username(void) {
+#ifdef ANDROID
+    char *name = NULL;
+    struct passwd *pwd = NULL;
+    pwd = getpwuid(getuid());
+    if ( pwd == NULL) {
+        return NULL;
+    }
+
+    name = strdup(pwd->pw_name);
+
+#else
     struct passwd pwd;
     struct passwd *pwdbuf;
     char buf[NSS_BUFLEN_PASSWD];
@@ -252,6 +264,7 @@
 
     name = strdup(pwd.pw_name);
 
+#endif
     if (name == NULL) {
       return NULL;
