--- a/ConfigureChecks.cmake	2014-12-27 04:30:13.000000000 -0600
+++ b/ConfigureChecks.cmake	2015-01-07 12:20:43.000000000 -0600
@@ -104,8 +104,14 @@
 check_function_exists(poll HAVE_POLL)
 check_function_exists(select HAVE_SELECT)
 check_function_exists(getaddrinfo HAVE_GETADDRINFO)
+
+if (APPLE)
+check_symbol_exists(ntohll sys/_endian.h HAVE_NTOHLL)
+check_symbol_exists(htonll sys/_endian.h HAVE_HTONLL)
+else (APPLE)
 check_function_exists(ntohll HAVE_NTOHLL)
 check_function_exists(htonll HAVE_HTONLL)
+endif(APPLE)
 
 if (WIN32)
     check_function_exists(_strtoui64 HAVE__STRTOUI64)
