--- include/libssh/wrapper.h.orig	2014-12-27 12:03:55.521114159 +0100
+++ include/libssh/wrapper.h	2014-12-27 12:06:35.072310640 +0100
@@ -52,7 +52,7 @@
 };
 
 typedef struct ssh_mac_ctx_struct *ssh_mac_ctx;
-MD5CTX md5_init(void);
+MD5CTX ssh_md5_init(void);
 void md5_update(MD5CTX c, const void *data, unsigned long len);
 void md5_final(unsigned char *md,MD5CTX c);
 
--- src/dh.c.orig	2014-12-27 12:08:07.441537435 +0100
+++ src/dh.c	2014-12-27 12:09:25.539205892 +0100
@@ -906,7 +906,7 @@
     return SSH_ERROR;
   }
 
-  ctx = md5_init();
+  ctx = ssh_md5_init();
   if (ctx == NULL) {
     SAFE_FREE(h);
     return SSH_ERROR;
@@ -1037,7 +1037,7 @@
                 goto out;
             }
 
-            ctx = md5_init();
+            ctx = ssh_md5_init();
             if (ctx == NULL) {
                 free(h);
                 rc = -1;
--- src/kex1.c.orig	2014-12-27 12:11:18.307842585 +0100
+++ src/kex1.c	2014-12-27 12:11:38.999225682 +0100
@@ -78,7 +78,7 @@
     ssh_string hostn) {
   MD5CTX md5 = NULL;
 
-  md5 = md5_init();
+  md5 = ssh_md5_init();
   if (md5 == NULL) {
     return -1;
   }
--- src/libcrypto.c.orig	2014-12-27 12:12:40.977378184 +0100
+++ src/libcrypto.c	2014-12-27 12:13:14.716372667 +0100
@@ -234,7 +234,7 @@
   SHA512(digest, len, hash);
 }
 
-MD5CTX md5_init(void) {
+MD5CTX ssh_md5_init(void) {
   MD5CTX c = malloc(sizeof(*c));
   if (c == NULL) {
     return NULL;
--- src/libgcrypt.c.orig	2014-12-27 12:13:46.319430927 +0100
+++ src/libgcrypt.c	2014-12-27 12:13:59.611034883 +0100
@@ -132,7 +132,7 @@
   gcry_md_hash_buffer(GCRY_MD_SHA512, hash, digest, len);
 }
 
-MD5CTX md5_init(void) {
+MD5CTX ssh_md5_init(void) {
   MD5CTX c = NULL;
   gcry_md_open(&c, GCRY_MD_MD5, 0);
 
