diff -ur openssl/Configure openssl/Configure
--- openssl/Configure	2009-02-19 05:43:18.000000000 -0400
+++ openssl/Configure	2009-04-02 15:39:19.000000000 -0400
@@ -1218,7 +1218,14 @@
 	$cflags = "-DZLIB $cflags";
 	if (defined($disabled{"zlib-dynamic"}))
 		{
-		$lflags = "$lflags -lz";
+		if (defined($withargs{"zlib-lib"}))
+			{
+			$lflags = "$lflags -L" . $withargs{"zlib-lib"} . " -lz";
+			}
+		else
+			{
+			$lflags = "$lflags -lz";
+			}
 		}
 	else
 		{
--- openssl//Makefile.org	2010-01-27 11:06:36.000000000 -0500
+++ openssl/Makefile.org	20102012-12-31 17:47:43.392861820 -0500
@@ -210,6 +210,7 @@
 		MAKEDEPPROG='${MAKEDEPPROG}'			\
 		SHARED_LDFLAGS='${SHARED_LDFLAGS}'		\
 		KRB5_INCLUDES='${KRB5_INCLUDES}' LIBKRB5='${LIBKRB5}'	\
+		ZLIB_INCLUDE='$(ZLIB_INCLUDE)' LIBZLIB='$(LIBZLIB)'     \
 		EXE_EXT='${EXE_EXT}' SHARED_LIBS='${SHARED_LIBS}'	\
 		SHLIB_EXT='${SHLIB_EXT}' SHLIB_TARGET='${SHLIB_TARGET}'	\
 		PEX_LIBS='${PEX_LIBS}' EX_LIBS='${EX_LIBS}'	\
--- openssl/crypto/Makefile	2008-09-17 13:10:55.000000000 -0400
+++ openssl/crypto/Makefile	2012-12-31 17:46:36.292861863 -0500
@@ -5,9 +5,9 @@
 DIR=		crypto
 TOP=		..
 CC=		cc
-INCLUDE=	-I. -I$(TOP) -I../include
+INCLUDE=	-I. -I$(TOP) -I../include $(ZLIB_INCLUDE)
 # INCLUDES targets sudbirs!
-INCLUDES=	-I.. -I../.. -I../../include
+INCLUDES=	-I.. -I../.. -I../../include $(ZLIB_INCLUDE)
 CFLAG=		-g
 MAKEDEPPROG=	makedepend
 MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
