--- a/meson.build
+++ b/meson.build
@@ -158,7 +158,7 @@
     cdata.set('HAVE_PTHREAD_H', 1)
 endif
 
-freetype_dependency = dependency('freetype2', required: get_option('freetype'))
+freetype_dependency = dependency('freetype2', 'freetype', required: get_option('freetype'))
 if freetype_dependency.found()
     cdata.set('HAVE_FT2', 1)
 endif
@@ -181,7 +181,7 @@
     endif
 endif
 
-libxml2_dependency = dependency('libxml-2.0', version: '>= 2.6', required: get_option('libxml2'))
+libxml2_dependency = dependency('libxml-2.0', 'libxml2', version: '>= 2.6', required: get_option('libxml2'))
 if libxml2_dependency.found()
     cdata.set('HAVE_LIBXML2', 1)
 endif
