 src/file/mount.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/file/mount.c b/src/file/mount.c
index 6382d8b..4f5392a 100644
--- a/src/file/mount.c
+++ b/src/file/mount.c
@@ -44,6 +44,7 @@ char *mount_get_mountpoint(const char *device_path)
         return str_dup(device_path);
     }
 
+#ifndef ANDROID
     FILE *f = setmntent ("/proc/self/mounts", "r");
     if (f) {
         struct mntent* m;
@@ -61,6 +62,7 @@ char *mount_get_mountpoint(const char *device_path)
         }
         endmntent (f);
     }
+#endif
 #endif /* HAVE_MNTENT_H */
 
     return str_dup(device_path);
