--- a/src/libbluray/bdj/bdj.c
+++ b/src/libbluray/bdj/bdj.c
@@ -252,7 +252,7 @@
         return NULL;
     }
 
-    switch (java_home_pid = vfork())
+    /*switch (java_home_pid = vfork())
     {
         case -1:
             BD_DEBUG(DBG_BDJ | DBG_CRIT, "vfork failed\n");
@@ -284,7 +284,7 @@
 
             waitpid(java_home_pid, &exitcode, 0);
     }
-
+*/
     if (result[0] == '\0' || exitcode) {
         BD_DEBUG(DBG_BDJ | DBG_CRIT,
                  "Unable to read path from " MACOS_JAVA_HOME "\n");
