Index: Renderer.hpp
===================================================================
--- Renderer.hpp	(revision 1028)
+++ Renderer.hpp	(working copy)
@@ -55,7 +55,7 @@
   int drawtitle;
   int texsize;
 
-  Renderer( int width, int height, int gx, int gy, int texsize,  BeatDetect *beatDetect, std::string presetURL, std::string title_fontURL, std::string menu_fontURL);
+  Renderer( int width, int height, int gx, int gy, int texsize,  BeatDetect *beatDetect, std::string presetURL, std::string title_fontURL, std::string menu_fontURL, int xpos, int ypos);
   ~Renderer();
   void RenderFrame(PresetOutputs *presetOutputs, PresetInputs *presetInputs);
   void ResetTextures();
@@ -90,7 +90,9 @@
   int gy;
 
   std::string m_presetName;
-  
+
+  int vx;
+  int vy;
   int vw; 
   int vh;
   
Index: projectM.cpp
===================================================================
--- projectM.cpp	(revision 1028)
+++ projectM.cpp	(working copy)
@@ -149,6 +149,8 @@
 	config.add("Aspect Correction", settings.aspectCorrection);
 	config.add("Easter Egg Parameter", settings.easterEgg);
 	config.add("Shuffle Enabled", settings.shuffleEnabled);
+	config.add("Window Left", settings.windowLeft);
+	config.add("Window Bottom", settings.windowBottom);
 	
 	std::fstream file(configFile.c_str());	
 	if (file) {
@@ -171,6 +173,8 @@
 	_settings.fps = config.read<int> ( "FPS", 35 );
 	_settings.windowWidth  = config.read<int> ( "Window Width", 512 );
 	_settings.windowHeight = config.read<int> ( "Window Height", 512 );
+	_settings.windowLeft  = config.read<int> ( "Window Left", 0 );
+	_settings.windowBottom = config.read<int> ( "Window Bottom", 0 );
 	_settings.smoothPresetDuration =  config.read<int> 
 			( "Smooth Preset Duration", config.read<int>("Smooth Transition Duration", 10));
 	_settings.presetDuration = config.read<int> ( "Preset Duration", 15 );
@@ -216,7 +220,8 @@
 	
 	
 	 projectM_init ( _settings.meshX, _settings.meshY, _settings.fps,
-			 _settings.textureSize, _settings.windowWidth,_settings.windowHeight);
+			 _settings.textureSize, _settings.windowWidth,_settings.windowHeight,
+			 _settings.windowLeft, _settings.windowBottom);
 
 	
 	 _settings.beatSensitivity = beatDetect->beat_sensitivity = config.read<float> ( "Hard Cut Sensitivity", 10.0 );
@@ -422,7 +427,7 @@
 	projectM_resetengine();
 }
 
-void projectM::projectM_init ( int gx, int gy, int fps, int texsize, int width, int height )
+void projectM::projectM_init ( int gx, int gy, int fps, int texsize, int width, int height, int xpos, int ypos )
 {
 
 	/** Initialise engine variables */
@@ -462,7 +467,7 @@
 	this->presetInputs2.gx = gx;
 	this->presetInputs2.gy = gy;
 
-	this->renderer = new Renderer ( width, height, gx, gy, texsize,  beatDetect, settings().presetURL, settings().titleFontURL, settings().menuFontURL );
+	this->renderer = new Renderer ( width, height, gx, gy, texsize,  beatDetect, settings().presetURL, settings().titleFontURL, settings().menuFontURL, xpos, ypos);
 	
 	running = true;
 
Index: Renderer.cpp
===================================================================
--- Renderer.cpp	(revision 1028)
+++ Renderer.cpp	(working copy)
@@ -11,7 +11,7 @@
 
 class Preset;
 
-Renderer::Renderer(int width, int height, int gx, int gy, int texsize, BeatDetect *beatDetect, std::string _presetURL, std::string _titlefontURL, std::string _menufontURL): title_fontURL(_titlefontURL), menu_fontURL(_menufontURL), presetURL(_presetURL), m_presetName("None"), vw(width), vh(height), gx(gx), gy(gy), texsize(texsize)
+Renderer::Renderer(int width, int height, int gx, int gy, int texsize, BeatDetect *beatDetect, std::string _presetURL, std::string _titlefontURL, std::string _menufontURL, int xpos, int ypos): title_fontURL(_titlefontURL), menu_fontURL(_menufontURL), presetURL(_presetURL), m_presetName("None"), vw(width), vh(height), gx(gx), gy(gy), texsize(texsize), vx(xpos), vy(ypos)
 {
 	int x; int y;
 	
@@ -217,7 +217,7 @@
 	}
 	else 
 #endif
-	glViewport( 0, 0, this->vw, this->vh );
+	glViewport( vx, vy, this->vw + vx, this->vh + vy );
 	
 	
 	
Index: projectM.hpp
===================================================================
--- projectM.hpp	(revision 1028)
+++ projectM.hpp	(working copy)
@@ -141,6 +141,8 @@
 	int textureSize;
 	int windowWidth;
 	int windowHeight;
+	int windowLeft;
+	int windowBottom;
 	std::string presetURL;
 	std::string titleFontURL;
 	std::string menuFontURL;		
@@ -265,7 +267,7 @@
   
   void switchPreset(std::unique_ptr<Preset> & targetPreset, PresetInputs & inputs, PresetOutputs & outputs);
   void readConfig(const std::string & configFile);
-  void projectM_init(int gx, int gy, int fps, int texsize, int width, int height);
+  void projectM_init(int gx, int gy, int fps, int texsize, int width, int height, int xpos, int ypos);
   void projectM_reset();
 
   void projectM_initengine();
