diff -Naur tvheadend-orig/src/htsp.c tvheadend-patch/src/htsp.c
--- tvheadend-orig/src/htsp.c	2010-11-06 14:37:15.015505279 +0100
+++ tvheadend-patch/src/htsp.c	2010-11-06 14:45:50.311506646 +0100
@@ -297,6 +297,7 @@
   channel_tag_mapping_t *ctm;
   channel_tag_t *ct;
   th_transport_t *t;
+  uint16_t caid = 0;
 
   htsmsg_t *out = htsmsg_create_map();
   htsmsg_t *tags = htsmsg_create_list();
@@ -323,8 +324,13 @@
     htsmsg_add_str(svcmsg, "name", transport_nicename(t));
     htsmsg_add_str(svcmsg, "type", transport_servicetype_txt(t));
     htsmsg_add_msg(services, NULL, svcmsg);
+
+    if (caid == 0) {
+    	caid = get_encryption_from_transport(t);
+    }
   }
 
+  htsmsg_add_u32(out, "channelCaid", caid);
   htsmsg_add_msg(out, "services", services);
   htsmsg_add_msg(out, "tags", tags);
   htsmsg_add_str(out, "method", method);
diff -Naur tvheadend-orig/src/transports.c tvheadend-patch/src/transports.c
--- tvheadend-orig/src/transports.c	2010-11-06 14:37:15.007505276 +0100
+++ tvheadend-patch/src/transports.c	2010-11-06 14:46:40.579505415 +0100
@@ -1025,3 +1025,32 @@
   if(t->tht_ch != NULL)
     htsp_channel_update(t->tht_ch);
 }
+
+/**
+ * Get the encryption CAID from a transport
+ *
+ * only the first CA stream in a transport is returned
+ */
+uint16_t
+get_encryption_from_transport(th_transport_t *t)
+{
+  th_stream_t *st;
+  caid_t *c;
+
+  TAILQ_FOREACH(st, &t->tht_components, st_link) {
+    switch(st->st_type) {
+      case SCT_CA:
+    	while((c = LIST_FIRST(&st->st_caids)) != NULL) {
+          if (c->caid) {
+            return c->caid;
+          }
+    	}
+    	break;
+      default:
+        break;
+	}
+  }
+
+  return 0;
+}
+
diff -Naur tvheadend-orig/src/transports.h tvheadend-patch/src/transports.h
--- tvheadend-orig/src/transports.h	2010-11-06 14:37:15.007505276 +0100
+++ tvheadend-patch/src/transports.h	2010-11-06 14:45:50.311506646 +0100
@@ -98,4 +98,6 @@
 
 int tss2errcode(int tss);
 
+uint16_t get_encryption_from_transport(th_transport_t *t);
+
 #endif /* TRANSPORTS_H */
