--- a/lib/libUPnP/Neptune/Source/System/Win32/NptWin32Threads.cpp
+++ b/lib/libUPnP/Neptune/Source/System/Win32/NptWin32Threads.cpp
@@ -636,10 +636,10 @@
     } else {
         return NPT_SUCCESS;
     }
-|   NPT_Win32Thread::CancelBlockerSocket
 }
 
 /*----------------------------------------------------------------------
+|   NPT_Win32Thread::CancelBlockerSocket
 +---------------------------------------------------------------------*/
 NPT_Result
 NPT_Win32Thread::CancelBlockerSocket()
