From 290644f4063539698fc27e0e8ba9b9c8a9f9f0b4 Mon Sep 17 00:00:00 2001
From: montellese <montellese@xbmc.org>
Date: Tue, 13 May 2014 21:31:41 +0200
Subject: [PATCH 24/24] platinum: enable logging

---
 lib/libUPnP/Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/libUPnP/Makefile.in b/lib/libUPnP/Makefile.in
index 03dfdd8..75cc448 100644
--- a/lib/libUPnP/Makefile.in
+++ b/lib/libUPnP/Makefile.in
@@ -99,6 +99,8 @@ endif
 
 INCLUDES=-I./Neptune/Source/Core -I./Platinum/Source/Core -I./Platinum/Source/Platinum -I./Platinum/Source/Devices/MediaConnect -I./Platinum/Source/Devices/MediaRenderer -I./Platinum/Source/Devices/MediaServer -I./Neptune/Source/System/Posix -I./Platinum/Source/Extras
 
+CXXFLAGS+=-DNPT_CONFIG_ENABLE_LOGGING
+
 LIB=libupnp.a
 
 include ../../Makefile.include
-- 
1.7.11.msysgit.0

