diff --git a/lib/libUPnP/Platinum/Source/Apps/FrameStreamer/main.cpp b/lib/libUPnP/Platinum/Source/Apps/FrameStreamer/main.cpp
index fba4986830..9956b6370b 100644
--- a/lib/libUPnP/Platinum/Source/Apps/FrameStreamer/main.cpp
+++ b/lib/libUPnP/Platinum/Source/Apps/FrameStreamer/main.cpp
@@ -235,8 +235,8 @@ main(int argc, char** argv)
         return 1;
 
     char buf[256];
-    while (gets(buf))
-    {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
         {
             break;
diff --git a/lib/libUPnP/Platinum/Source/Apps/MediaConnect/main.cpp b/lib/libUPnP/Platinum/Source/Apps/MediaConnect/main.cpp
index 993005f35e..97f1a72525 100644
--- a/lib/libUPnP/Platinum/Source/Apps/MediaConnect/main.cpp
+++ b/lib/libUPnP/Platinum/Source/Apps/MediaConnect/main.cpp
@@ -118,8 +118,8 @@ main(int argc, char** argv)
         return 1;
 
     char buf[256];
-    while (gets(buf))
-    {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
         {
             break;
diff --git a/lib/libUPnP/Platinum/Source/Apps/MediaCrawler/main.cpp b/lib/libUPnP/Platinum/Source/Apps/MediaCrawler/main.cpp
index 80a1a2323b..cb57f61337 100644
--- a/lib/libUPnP/Platinum/Source/Apps/MediaCrawler/main.cpp
+++ b/lib/libUPnP/Platinum/Source/Apps/MediaCrawler/main.cpp
@@ -71,7 +71,8 @@ int main(void)
     ctrlPoint->Discover(NPT_HttpUrl("255.255.255.255", 1900, "*"), "upnp:rootdevice", 1);
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/lib/libUPnP/Platinum/Source/Core/PltHttpServer.cpp b/lib/libUPnP/Platinum/Source/Core/PltHttpServer.cpp
index 2557f62851..777705702c 100644
--- a/lib/libUPnP/Platinum/Source/Core/PltHttpServer.cpp
+++ b/lib/libUPnP/Platinum/Source/Core/PltHttpServer.cpp
@@ -190,7 +190,7 @@ PLT_HttpServer::ServeFile(const NPT_HttpRequest&        request,
     NPT_FileInfo             file_info;
     
     // prevent hackers from accessing files outside of our root
-    if ((file_path.Find("/..") >= 0) || (file_path.Find("\\..") >= 0)) {
+    if ((file_path.Find("../") >= 0) || (file_path.Find("..\\") >= 0)) {
         return NPT_ERROR_NO_SUCH_ITEM;
     }
     
diff --git a/lib/libUPnP/Platinum/Source/Tests/FileMediaServer/FileMediaServerTest.cpp b/lib/libUPnP/Platinum/Source/Tests/FileMediaServer/FileMediaServerTest.cpp
index 0d1223baa4..3aca9704cd 100644
--- a/lib/libUPnP/Platinum/Source/Tests/FileMediaServer/FileMediaServerTest.cpp
+++ b/lib/libUPnP/Platinum/Source/Tests/FileMediaServer/FileMediaServerTest.cpp
@@ -148,7 +148,8 @@ main(int /* argc */, char** argv)
     NPT_LOG_INFO("Press 'q' to quit.");
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/lib/libUPnP/Platinum/Source/Tests/LightSample/LightSampleTest.cpp b/lib/libUPnP/Platinum/Source/Tests/LightSample/LightSampleTest.cpp
index d9f35a0a93..1ca3010916 100644
--- a/lib/libUPnP/Platinum/Source/Tests/LightSample/LightSampleTest.cpp
+++ b/lib/libUPnP/Platinum/Source/Tests/LightSample/LightSampleTest.cpp
@@ -65,7 +65,8 @@ main(int /* argc */, char** /* argv */)
     upnp.Start();
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/lib/libUPnP/Platinum/Source/Tests/MediaRenderer/MediaRendererTest.cpp b/lib/libUPnP/Platinum/Source/Tests/MediaRenderer/MediaRendererTest.cpp
index 54d9d24eb0..2e9d0f5e07 100644
--- a/lib/libUPnP/Platinum/Source/Tests/MediaRenderer/MediaRendererTest.cpp
+++ b/lib/libUPnP/Platinum/Source/Tests/MediaRenderer/MediaRendererTest.cpp
@@ -102,7 +102,8 @@ main(int /* argc */, char** argv)
     upnp.Start();
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
 
diff --git a/lib/libUPnP/Platinum/Source/Tests/Simple/SimpleTest.cpp b/lib/libUPnP/Platinum/Source/Tests/Simple/SimpleTest.cpp
index 6596227951..a4ca1c5356 100644
--- a/lib/libUPnP/Platinum/Source/Tests/Simple/SimpleTest.cpp
+++ b/lib/libUPnP/Platinum/Source/Tests/Simple/SimpleTest.cpp
@@ -52,7 +52,8 @@ main(int /* argc */, char** /* argv */)
     upnp.Start();
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/lib/libUPnP/Platinum/Source/Tests/Ssdp/SsdpTest1.cpp b/lib/libUPnP/Platinum/Source/Tests/Ssdp/SsdpTest1.cpp
index 1e1f6aa42e..9e8be1abe5 100644
--- a/lib/libUPnP/Platinum/Source/Tests/Ssdp/SsdpTest1.cpp
+++ b/lib/libUPnP/Platinum/Source/Tests/Ssdp/SsdpTest1.cpp
@@ -55,7 +55,8 @@ main(int, char**)
         return 1;
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/lib/libUPnP/Platinum/Source/Tools/SsdpProxy/SsdpProxy.cpp b/lib/libUPnP/Platinum/Source/Tools/SsdpProxy/SsdpProxy.cpp
index a0643a9a84..e487b6f409 100644
--- a/lib/libUPnP/Platinum/Source/Tools/SsdpProxy/SsdpProxy.cpp
+++ b/lib/libUPnP/Platinum/Source/Tools/SsdpProxy/SsdpProxy.cpp
@@ -396,7 +396,8 @@ main(int argc, char** argv)
     fprintf(stdout, "Enter q to quit\n");
 
     char buf[256];
-    while (gets(buf)) {
+    while (true) {
+        fgets(buf, 256, stdin);
         if (*buf == 'q')
             break;
     }
diff --git a/xbmc/network/upnp/UPnPRenderer.cpp b/xbmc/network/upnp/UPnPRenderer.cpp
index 96f020dbfd..4ad1774ca2 100644
--- a/xbmc/network/upnp/UPnPRenderer.cpp
+++ b/xbmc/network/upnp/UPnPRenderer.cpp
@@ -205,7 +205,7 @@ CUPnPRenderer::ProcessHttpGetRequest(NPT_HttpRequest&              request,
             }
 
             // prevent hackers from accessing files outside of our root
-            if ((filepath.Find("/..") >= 0) || (filepath.Find("\\..") >=0)) {
+            if ((filepath.Find("../") >= 0) || (filepath.Find("..\\") >= 0) {
                 return NPT_FAILURE;
             }
 
