diff --git a/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp b/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
index d22dc02bcf..d49bcec4e7 100644
--- a/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
+++ b/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
@@ -227,16 +227,27 @@ PLT_ProtocolInfo::ValidateField(const char*  val,
     if (num_chars && NPT_StringLength(val) != num_chars)
         return NPT_ERROR_INVALID_SYNTAX;
 
-    while (val) {
-        char c = *val++;
-        if (c == '\0') return NPT_SUCCESS;
+    while (val)
+    {
+        const char c = *val;
+        if (c == '\0')
+        {
+            return NPT_SUCCESS;
+        }
+        ++val;
 
         // look for character in valid chars
         const char* p = valid_chars;
-        while (*p != c && ++p) {};
+        while ((*p != c) && (*p != '\0'))
+        {
+            ++p;
+        }
 
         // reached end of valid chars means we didn't find it
-        if (!p) break;
+        if (*p == '\0')
+        {
+            break;
+        }
     }
 
     return NPT_ERROR_INVALID_SYNTAX;
