--- Lib/urllib.py	2010-02-20 17:13:01.000000000 -0500
+++ Lib/urllib.py	2013-03-14 18:12:02.526251189 -0400
@@ -1330,7 +1330,6 @@
 
 
 if sys.platform == 'darwin':
-    from _scproxy import _get_proxy_settings, _get_proxies
 
     def proxy_bypass_macosx_sysconf(host):
         """
@@ -1352,7 +1351,7 @@
                 parts = (parts + [0, 0, 0, 0])[:4]
             return (parts[0] << 24) | (parts[1] << 16) | (parts[2] << 8) | parts[3]
 
-        proxy_settings = _get_proxy_settings()
+        return False
 
         # Check for simple host names:
         if '.' not in host:
@@ -1393,7 +1392,7 @@
         This function uses the MacOSX framework SystemConfiguration
         to fetch the proxy information.
         """
-        return _get_proxies()
+        return {}
 
 
 
