--- Mac/Modules/_scproxy.c	2009-09-20 15:21:24.000000000 -0400
+++ Mac/Modules/_scproxy.c	2013-01-22 19:40:41.000000000 -0500
@@ -4,7 +4,6 @@
  */
 #include <Python.h>
 #include <SystemConfiguration/SystemConfiguration.h>
-
 static int32_t 
 cfnum_to_int32(CFNumberRef num)
 {
@@ -48,6 +47,7 @@
 static PyObject*
 get_proxy_settings(PyObject* mod __attribute__((__unused__)))
 {
+/*
 	CFDictionaryRef proxyDict = NULL;
 	CFNumberRef aNum = NULL;
 	CFArrayRef anArray = NULL;
@@ -55,7 +55,7 @@
 	PyObject* v;
 	int r;
 
-	proxyDict = SCDynamicStoreCopyProxies(NULL);
+	proxyDict =  CFNetworkCopyProxiesForURL(NULL);
 	if (!proxyDict) {
 		Py_INCREF(Py_None);
 		return Py_None;
@@ -114,6 +114,7 @@
 error:
 	if (proxyDict)  CFRelease(proxyDict);
 	Py_XDECREF(result);
+*/
 	return NULL;
 }
 
@@ -163,6 +164,7 @@
 static PyObject*
 get_proxies(PyObject* mod __attribute__((__unused__)))
 {
+/*
 	PyObject* result = NULL;
 	int r;
 	CFDictionaryRef proxyDict = NULL;
@@ -201,6 +203,7 @@
 error:
 	if (proxyDict)  CFRelease(proxyDict);
 	Py_XDECREF(result);
+*/
 	return NULL;
 }
 
