--- a/src/libcec/cmake/CheckPlatformSupport.cmake
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake
@@ -40,7 +40,7 @@
 if(WIN32)
   # Windows
   add_definitions(-DTARGET_WINDOWS -DNOMINMAX -D_CRT_SECURE_NO_WARNINGS -D_WINSOCKAPI_)
-  set(LIB_DESTINATION ".")
+  set(LIB_DESTINATION "${CMAKE_INSTALL_BINDIR}")

   set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")

--- a/src/libcec/CMakeLists.txt
+++ b/src/libcec/CMakeLists.txt
@@ -250,7 +250,7 @@
 install(TARGETS ${CEC_INSTALL_TARGETS}
         EXPORT libcec
         RUNTIME DESTINATION ${LIB_DESTINATION}
-        ARCHIVE DESTINATION ${LIB_DESTINATION}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         LIBRARY DESTINATION ${LIB_DESTINATION})

 include(CMakePackageConfigHelpers)
