--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@
 set(LIBCEC_VERSION_PATCH 0)
 
 # cec-client
+if(ENABLE_CLIENT)
 add_subdirectory(src/cec-client)
 add_dependencies(cec-client cec)
 
@@ -15,6 +16,7 @@
 
 # pyCecClient
 add_subdirectory(src/pyCecClient)
+endif()
 
 # libCEC
 add_subdirectory(src/libcec)
@@ -25,6 +27,7 @@
 
 # windows specific files
 if(WIN32)
+if(ENABLE_CLIENT)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/project/nsis/libcec-version.nsh.in
                  ${CMAKE_CURRENT_SOURCE_DIR}/project/nsis/libcec-version.nsh)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/dotnetlib/LibCecSharp/LibCecSharp.rc.in
@@ -44,3 +47,4 @@
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/dotnet/src/CecSharpTester/netfx/Properties/AssemblyInfo.cs.in
                  ${CMAKE_CURRENT_SOURCE_DIR}/src/dotnet/src/CecSharpTester/netfx/Properties/AssemblyInfo.cs)
 endif()
+endif()
