--- a/src/libcec/CMakeLists.txt
+++ b/src/libcec/CMakeLists.txt
@@ -175,7 +175,7 @@
                         ${CMAKE_INSTALL_PREFIX}/include)
 
   install(FILES         ${CMAKE_CURRENT_SOURCE_DIR}/libcec.pc
-          DESTINATION   ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+          DESTINATION   lib/pkgconfig)
 endif()
 
 # install headers
@@ -193,11 +193,25 @@
 # libCEC shared target
 add_library(cec SHARED ${CEC_SOURCES})
 install(TARGETS cec
-        DESTINATION ${LIB_DESTINATION})
+        EXPORT libcec
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib
+        LIBRARY DESTINATION lib)
+
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/libcec-config.cmake
+        DESTINATION lib/cmake/libcec)
+
 set_target_properties(cec PROPERTIES VERSION   ${LIBCEC_VERSION_MAJOR}.${LIBCEC_VERSION_MINOR}.${LIBCEC_VERSION_PATCH}
                                      SOVERSION ${LIBCEC_VERSION_MAJOR})
 target_link_libraries(cec ${cec_depends})
 
+install(EXPORT libcec
+  NAMESPACE
+    libcec::
+  DESTINATION
+    lib/cmake/libcec
+)
+
 if (MSVC)
   # generate pdb in release mode too
   set_target_properties(cec
--- /dev/null
+++ b/src/libcec/cmake/libcec-config.cmake
@@ -0,0 +1 @@
+include(${CMAKE_CURRENT_LIST_DIR}/libcec.cmake)
