--- a/src/libcec/cmake/SetBuildInfo.cmake
+++ b/src/libcec/cmake/SetBuildInfo.cmake
@@ -14,6 +14,7 @@
   set(LIB_INFO "")
 
   # add git revision to compile info
+  if(0)
   find_program(HAVE_GIT_BIN git /bin /usr/bin /usr/local/bin)
   if(HAVE_GIT_BIN)
     exec_program(${CMAKE_CURRENT_SOURCE_DIR}/cmake/git-rev.sh HEAD OUTPUT_VARIABLE GIT_REVISION)
@@ -22,6 +23,7 @@
   if (GIT_REVISION)
     set(LIB_INFO "git revision: ${GIT_REVISION},")
   endif()
+  endif()
 
   # add compilation date to compile info
   STRING(TIMESTAMP BUILD_DATE "%Y-%m-%d %H:%M:%S" UTC)
