--- a/src/libcec/cmake/SetBuildInfo.cmake
+++ b/src/libcec/cmake/SetBuildInfo.cmake
@@ -12,7 +12,7 @@
 else()
   # not Windows
   set(LIB_INFO "")
-
+if(NOT DISABLE_BUILDINFO)
   # add git revision to compile info
   find_program(HAVE_GIT_BIN git /bin /usr/bin /usr/local/bin)
   if(HAVE_GIT_BIN)
@@ -69,6 +69,6 @@
 
     set(LIB_INFO "${LIB_INFO} on ${BUILD_SYSNAME} ${BUILD_SYSVER} (${BUILD_SYSARCH})")
   endif()
-
 endif()
+endif()
 
