From ac7ee2f4aeabf713a81c1e11971dbe271fa2400c Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 02:15:41 -0400
Subject: [PATCH 10/32] additional HDMV types and reg descriptors

---
 libavformat/mpegts.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/libavformat/mpegts.c b/libavformat/mpegts.c
index b78f4a4..e17bc85 100644
--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -510,6 +510,8 @@ static const StreamType HDMV_types[] = {
     { 0x82, AVMEDIA_TYPE_AUDIO, CODEC_ID_DTS },
     { 0x83, AVMEDIA_TYPE_AUDIO, CODEC_ID_TRUEHD },
     { 0x84, AVMEDIA_TYPE_AUDIO, CODEC_ID_EAC3 },
+    { 0x85, CODEC_TYPE_AUDIO,     CODEC_ID_DTS }, /* DTS HD */
+    { 0x86, CODEC_TYPE_AUDIO,     CODEC_ID_DTS }, /* DTS HD MASTER*/
     { 0x90, AVMEDIA_TYPE_SUBTITLE, CODEC_ID_HDMV_PGS_SUBTITLE },
     { 0 },
 };
@@ -524,6 +526,10 @@ static const StreamType MISC_types[] = {
 static const StreamType REGD_types[] = {
     { MKTAG('d','r','a','c'), AVMEDIA_TYPE_VIDEO, CODEC_ID_DIRAC },
     { MKTAG('A','C','-','3'), AVMEDIA_TYPE_AUDIO,   CODEC_ID_AC3 },
+    { MKTAG('D','T','S','1'), CODEC_TYPE_AUDIO,   CODEC_ID_DTS },
+    { MKTAG('D','T','S','2'), CODEC_TYPE_AUDIO,   CODEC_ID_DTS },
+    { MKTAG('D','T','S','3'), CODEC_TYPE_AUDIO,   CODEC_ID_DTS },
+    { MKTAG('V','C','-','1'), CODEC_TYPE_VIDEO,   CODEC_ID_VC1 },
     { 0 },
 };
 
-- 
1.7.0

