From 8a30526df1c047d946d015167fbde54a5696384e Mon Sep 17 00:00:00 2001
From: Anssi Hannula <anssi.hannula@iki.fi>
Date: Fri, 21 Jan 2011 20:45:56 +0200
Subject: [PATCH 45/46] libfaac: add recognized profiles array

Submitted upstream.

---
 libavcodec/libfaac.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/libavcodec/libfaac.c b/libavcodec/libfaac.c
index b220b17..b1bf54d 100644
--- a/libavcodec/libfaac.c
+++ b/libavcodec/libfaac.c
@@ -145,6 +145,14 @@ static av_cold int Faac_encode_close(AVCodecContext *avctx)
     return 0;
 }
 
+static const AVProfile profiles[] = {
+    { FF_PROFILE_AAC_MAIN, "Main" },
+    { FF_PROFILE_AAC_LOW,  "Low"  },
+    { FF_PROFILE_AAC_SSR,  "SSR"  },
+    { FF_PROFILE_AAC_LTP,  "LTP"  },
+    { FF_PROFILE_UNKNOWN },
+};
+
 AVCodec libfaac_encoder = {
     "libfaac",
     AVMEDIA_TYPE_AUDIO,
@@ -155,4 +163,5 @@ AVCodec libfaac_encoder = {
     Faac_encode_close,
     .sample_fmts = (const enum AVSampleFormat[]){AV_SAMPLE_FMT_S16,AV_SAMPLE_FMT_NONE},
     .long_name = NULL_IF_CONFIG_SMALL("libfaac AAC (Advanced Audio Codec)"),
+    .profiles = profiles,
 };
-- 
1.7.3

