From e2577429a6cfcce7554ce9072365308688787598 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sat, 10 Apr 2021 08:16:11 -0700
Subject: [PATCH 1/4] ffmpeg: windows: configure: detect openssl

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index ffa407d53d..c4fe14568a 100755
--- a/configure
+++ b/configure
@@ -7106,6 +7106,8 @@ enabled openssl           && { { check_pkg_config openssl "openssl >= 3.0.0" ope
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto ||
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl32 -leay32 ||
                                check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||
+                               check_lib openssl openssl/ssl.h OPENSSL_init_ssl -llibssl -llibcrypto -lws2_32 -lgdi32 -ladvapi32 -luser32 ||
+                               check_lib openssl openssl/ssl.h SSL_library_init -llibeay32 -lssleay32 ||
                                die "ERROR: openssl not found"; }
 enabled pocketsphinx      && require_pkg_config pocketsphinx pocketsphinx pocketsphinx/pocketsphinx.h ps_init
 enabled rkmpp             && { require_pkg_config rkmpp rockchip_mpp  rockchip/rk_mpi.h mpp_create &&
-- 
2.46.1

