diff -ur ffmpeg/configure ffmpeg-patch/configure
--- ffmpeg/configure	2020-01-31 13:06:00.000000000 +0000
+++ ffmpeg-patch/configure	2020-01-31 13:06:40.000000000 +0000
@@ -7439,6 +7439,9 @@
                                      $CONFIG_EXTRA      \
                                      $ALL_COMPONENTS    \
 
+echo "#if defined(HAVE_UNISTD_H) && HAVE_UNISTD_H == 0" >> $TMPH
+echo "#undef HAVE_UNISTD_H" >> $TMPH
+echo "#endif" >> $TMPH
 echo "#endif /* FFMPEG_CONFIG_H */" >> $TMPH
 echo "endif # FFMPEG_CONFIG_MAK" >> ffbuild/config.mak
 
