--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -3991,7 +3991,7 @@
     slave_fd = open(slave_name, O_RDWR | O_NOCTTY); /* open slave */
     if (slave_fd < 0)
         return posix_error();
-#if !defined(__CYGWIN__) && !defined(HAVE_DEV_PTC)
+#if !defined(__CYGWIN__) && !defined(HAVE_DEV_PTC) && !defined(ANDROID)
     ioctl(slave_fd, I_PUSH, "ptem"); /* push ptem */
     ioctl(slave_fd, I_PUSH, "ldterm"); /* push ldterm */
 #ifndef __hpux
--- a/Modules/pwdmodule.c
+++ b/Modules/pwdmodule.c
@@ -68,14 +68,14 @@
 #define SETS(i,val) sets(v, i, val)
 
     SETS(setIndex++, p->pw_name);
-#ifdef __VMS
+#if defined __VMS || defined(ANDROID)
     SETS(setIndex++, "");
 #else
     SETS(setIndex++, p->pw_passwd);
 #endif
     PyStructSequence_SET_ITEM(v, setIndex++, _PyInt_FromUid(p->pw_uid));
     PyStructSequence_SET_ITEM(v, setIndex++, _PyInt_FromGid(p->pw_gid));
-#ifdef __VMS
+#if defined __VMS || defined(ANDROID)
     SETS(setIndex++, "");
 #else
     SETS(setIndex++, p->pw_gecos);
--- a/Modules/termios.c
+++ b/Modules/termios.c
@@ -230,6 +230,7 @@
     return Py_None;
 }
 
+#if !defined(ANDROID)
 PyDoc_STRVAR(termios_tcdrain__doc__,
 "tcdrain(fd) -> None\n\
 \n\
@@ -249,6 +250,7 @@
     Py_INCREF(Py_None);
     return Py_None;
 }
+#endif
 
 PyDoc_STRVAR(termios_tcflush__doc__,
 "tcflush(fd, queue) -> None\n\
@@ -304,8 +306,10 @@
      METH_VARARGS, termios_tcsetattr__doc__},
     {"tcsendbreak", termios_tcsendbreak,
      METH_VARARGS, termios_tcsendbreak__doc__},
+#if !defined(ANDROID)
     {"tcdrain", termios_tcdrain,
      METH_VARARGS, termios_tcdrain__doc__},
+#endif
     {"tcflush", termios_tcflush,
      METH_VARARGS, termios_tcflush__doc__},
     {"tcflow", termios_tcflow,
