--- a/Python/dynload_shlib.c
+++ b/Python/dynload_shlib.c
@@ -112,10 +112,6 @@
     dlopenflags = PyThreadState_GET()->interp->dlopenflags;
 #endif
 
-    if (Py_VerboseFlag)
-        PySys_WriteStderr("dlopen(\"%s\", %x);\n", pathname,
-                          dlopenflags);
-
 #ifdef __VMS
     /* VMS currently don't allow a pathname, use a logical name instead */
     /* Concatenate 'python_module_' and shortname */
@@ -125,8 +121,17 @@
     PyOS_snprintf(pathbuf, sizeof(pathbuf), "python_module_%-.200s",
                   shortname);
     pathname = pathbuf;
+#elif defined(ANDROID)
+    /* Android does not allow a pathname and wants lib*.so */
+    PyOS_snprintf(pathbuf, sizeof(pathbuf), "lib%-.200s.so", 
+                  shortname);
+    pathname = pathbuf;
 #endif
 
+    if (Py_VerboseFlag)
+        PySys_WriteStderr("dlopen(\"%s\", %x);\n", pathname, 
+                          dlopenflags);
+
     handle = dlopen(pathname, dlopenflags);
 
     if (handle == NULL) {
