--- a/configure.ac
+++ b/configure.ac
@@ -3088,7 +3088,7 @@
  getentropy \
  getpriority getresuid getresgid getpwent getspnam getspent getsid getwd \
  initgroups kill killpg lchmod lchown lstat mkfifo mknod mktime mmap \
- mremap nice pathconf pause plock poll pthread_init \
+ mremap nice nl_langinfo pathconf pause plock poll pthread_init \
  putenv readlink realpath \
  select sem_open sem_timedwait sem_getvalue sem_unlink setegid seteuid \
  setgid \
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -294,7 +294,7 @@
         overridden = 1;
     }
 
-#if defined(Py_USING_UNICODE) && defined(HAVE_LANGINFO_H) && defined(CODESET)
+#if defined(Py_USING_UNICODE) && defined(HAVE_LANGINFO_H) && defined(HAVE_NL_LANGINFO) && defined(CODESET)
     /* On Unix, set the file system encoding according to the
        user's preference, if the CODESET names a well-known
        Python codec, and Py_FileSystemDefaultEncoding isn't
