--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -8649,7 +8649,7 @@
 static PyObject *
 posix_abort(PyObject *self, PyObject *noargs)
 {
-    abort();
+    // abort();
     /*NOTREACHED*/
     Py_FatalError("abort() called from Python code didn't abort!");
     return NULL;
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -1706,7 +1706,7 @@
     DebugBreak();
 #endif
 #endif /* MS_WINDOWS */
-    abort();
+	// abort();
 }
 
 /* Clean up and exit */
