--- a/Lib/urllib.py
+++ b/Lib/urllib.py
@@ -1300,10 +1300,11 @@
 
 def quote_plus(s, safe=''):
     """Quote the query fragment of a URL; replacing ' ' with '+'"""
-    if ' ' in s:
-        s = quote(s, safe + ' ')
-        return s.replace(' ', '+')
-    return quote(s, safe)
+    if not s is None:
+		if ' ' in s:
+			s = quote(s, safe + ' ')
+			return s.replace(' ', '+')
+		return quote(s, safe)
 
 def urlencode(query, doseq=0):
     """Encode a sequence of two-element tuples or dictionary into a URL query string.

@@ -1437,7 +1437,6 @@
 
 
 if sys.platform == 'darwin':
-    from _scproxy import _get_proxy_settings, _get_proxies
 
     def proxy_bypass_macosx_sysconf(host):
         """
@@ -1459,7 +1458,7 @@
                 parts = (parts + [0, 0, 0, 0])[:4]
             return (parts[0] << 24) | (parts[1] << 16) | (parts[2] << 8) | parts[3]
 
-        proxy_settings = _get_proxy_settings()
+        return False
 
         # Check for simple host names:
         if '.' not in host:
@@ -1504,7 +1503,7 @@
         This function uses the MacOSX framework SystemConfiguration
         to fetch the proxy information.
         """
-        return _get_proxies()
+        return {}
 
     def proxy_bypass(host):
         """Return True, if a host should be bypassed.
