diff --git a/Lib/ssl.py b/Lib/ssl.py
index 5df6ed3..a871519 100644
--- a/Lib/ssl.py
+++ b/Lib/ssl.py
@@ -482,7 +482,7 @@ def _create_unverified_context(protocol=PROTOCOL_SSLv23, cert_reqs=None,
     return context
 
 # Used by http.client if no context is explicitly passed.
-_create_default_https_context = create_default_context
+_create_default_https_context = _create_unverified_context
 
 
 # Backwards compatibility alias, even though it's not a public name.

