diff -ruN libsamplerate-0.1.7/src/float_cast.h.org src/float_cast.h
--- libsamplerate-0.1.7/src/float_cast.h.org	2010-10-28 21:00:19.000000000 -0400
+++ libsamplerate-0.1.7/src/float_cast.h	2010-10-28 21:00:52.000000000 -0400
@@ -193,7 +193,7 @@
 		return res [1] ;
 	} /* double2int */
 
-#elif (defined (__MACH__) && defined (__APPLE__))
+#elif (defined (__MACH__) && defined (__APPLE__) && !defined(__arm__))
 
 	/* For Apple MacOSX. */
 

