diff -NaurwB streamdev/server/connectionVTP.c streamdev-patched/server/connectionVTP.c
--- streamdev/server/connectionVTP.c	2009-10-13 08:38:47.000000000 +0200
+++ streamdev-patched/server/connectionVTP.c	2010-01-17 19:08:39.000000000 +0100
@@ -40,6 +40,9 @@
 #if defined(USE_PARENTALRATING) || defined(PARENTALRATINGCONTENTVERSNUM)
 	enum eStates { Channel, Event, Title, Subtitle, Description, Vps, Content,
 	               EndEvent, EndChannel, EndEPG };
+#elif APIVERSNUM >= 10711
+	enum eStates { Channel, Event, Title, Subtitle, Description, Vps, Content, Rating,
+	               EndEvent, EndChannel, EndEPG };
 #else
 	enum eStates { Channel, Event, Title, Subtitle, Description, Vps, 
 	               EndEvent, EndChannel, EndEPG };
@@ -282,7 +285,7 @@
 		break;
 
 	case Vps:
-#if defined(USE_PARENTALRATING) || defined(PARENTALRATINGCONTENTVERSNUM)
+#if defined(USE_PARENTALRATING) || defined(PARENTALRATINGCONTENTVERSNUM) || APIVERSNUM >= 10711
 		m_State = Content;
 #else
 		m_State = EndEvent;
@@ -308,6 +311,25 @@
 		} else
 			return Next(Last);
 		break;
+#elif APIVERSNUM >= 10711
+	case Content:
+		m_State = Rating;
+		if (!isempty(m_Event->ContentToString(m_Event->Contents()))) {
+			char *copy = strdup(m_Event->ContentToString(m_Event->Contents()));
+			cString cpy(copy, true);
+			strreplace(copy, '\n', '|');
+			return m_Client->Respond(-215, "G %i %i %s", m_Event->Contents() & 0xF0, m_Event->Contents() & 0x0F, copy);
+		} else
+			return Next(Last);
+		break;
+
+	case Rating:
+		m_State = EndEvent;
+		if (m_Event->ParentalRating())
+			return m_Client->Respond(-215, "R %d", m_Event->ParentalRating());
+		else
+			return Next(Last);
+		break;
 #endif
 
 	case EndEvent:
