diff -NaurwB vdr-1.7.7/skins.c vdr-1.7.7-patched/skins.c
--- vdr-1.7.7/skins.c	2008-02-17 12:31:09.000000000 +0100
+++ vdr-1.7.7-patched/skins.c	2009-11-23 05:13:30.000000000 +0100
@@ -237,7 +237,11 @@
      }
   cSkinDisplay::Current()->SetMessage(Type, s);
   cSkinDisplay::Current()->Flush();
+#ifdef USE_STATUS_EXTENSION
+  cStatus::MsgOsdStatusMessage(Type, s);
+#else
   cStatus::MsgOsdStatusMessage(s);
+#endif
   eKeys k = kNone;
   if (Type != mtStatus) {
      k = Interface->Wait(Seconds);
@@ -248,7 +252,11 @@
         }
      else {
         cSkinDisplay::Current()->SetMessage(Type, NULL);
+#ifdef USE_STATUS_EXTENSION
+        cStatus::MsgOsdStatusMessage(Type, NULL);
+#else
         cStatus::MsgOsdStatusMessage(NULL);
+#endif
         }
      }
   else if (!s && displayMessage) {
diff -NaurwB vdr-1.7.7/status.c vdr-1.7.7-patched/status.c
--- vdr-1.7.7/status.c	2009-06-13 01:01:25.000000000 +0200
+++ vdr-1.7.7-patched/status.c	2009-11-23 04:49:48.000000000 +0100
@@ -97,11 +97,22 @@
       sm->OsdTitle(Title);
 }
 
+#ifdef USE_STATUS_EXTENSION
+void cStatus::MsgOsdStatusMessage(eMessageType type, const char *Message)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+  {
+      sm->OsdStatusMessage(type, Message);
+      sm->OsdStatusMessage(Message); // For comaptibilty
+  }
+}
+#else
 void cStatus::MsgOsdStatusMessage(const char *Message)
 {
   for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
       sm->OsdStatusMessage(Message);
 }
+#endif
 
 void cStatus::MsgOsdHelpKeys(const char *Red, const char *Green, const char *Yellow, const char *Blue)
 {
diff -NaurwB vdr-1.7.7/status.h vdr-1.7.7-patched/status.h
--- vdr-1.7.7/status.h	2009-06-13 01:01:25.000000000 +0200
+++ vdr-1.7.7-patched/status.h	2009-11-23 04:50:34.000000000 +0100
@@ -17,6 +17,10 @@
 #ifdef USE_PINPLUGIN
 #include "plugin.h"
 #endif /* PINPLUGIN */
+#ifdef USE_STATUS_EXTENSION
+#include "skins.h"
+#endif /* STATUS_EXTENSION */
+
 
 enum eTimerChange { tcMod, tcAdd, tcDel };
 #ifdef USE_STREAMDEVEXT
@@ -80,6 +84,11 @@
   virtual void OsdStatusMessage(const char *Message) {}
                // Message has been displayed in the status line of the menu.
                // If Message is NULL, the status line has been cleared.
+#ifdef USE_STATUS_EXTENSION
+  virtual void OsdStatusMessage(eMessageType type, const char *Message) {}
+               // Message has been displayed in the status line of the menu.
+               // If Message is NULL, the status line has been cleared.
+#endif
   virtual void OsdHelpKeys(const char *Red, const char *Green, const char *Yellow, const char *Blue) {}
                // The help keys have been set to the given values (may be NULL).
   virtual void OsdItem(const char *Text, int Index) {}
@@ -145,7 +154,11 @@
   static void MsgSetSubtitleTrack(int Index, const char * const *Tracks);
   static void MsgOsdClear(void);
   static void MsgOsdTitle(const char *Title);
+#ifdef USE_STATUS_EXTENSION
+  static void MsgOsdStatusMessage(eMessageType type, const char *Message);
+#else
   static void MsgOsdStatusMessage(const char *Message);
+#endif
   static void MsgOsdHelpKeys(const char *Red, const char *Green, const char *Yellow, const char *Blue);
   static void MsgOsdItem(const char *Text, int Index);
   static void MsgOsdCurrentItem(const char *Text);
diff -NaurwB vdr-1.7.7/Make.config.template vdr-1.7.7-patched/Make.config.template
--- vdr-1.7.7/Make.config.template	2009-06-13 01:05:04.000000000 +0200
+++ vdr-1.7.7-patched/Make.config.template	2009-11-23 04:51:28.000000000 +0100
@@ -87,6 +87,7 @@
 #SOFTOSD = 1
 #SOURCECAPS = 1
 #SORTRECORDS = 1
+STATUS_EXTENSION = 1
 #STREAMDEVEXT = 1
 #TIMERCMD = 1
 #TIMERINFO = 1
@@ -240,6 +241,10 @@
 DEFINES += -DUSE_SETTIME
 endif
 
+ifdef STATUS_EXTENSION
+DEFINES += -DUSE_STATUS_EXTENSION
+endif
+
 ifdef SOFTOSD
 DEFINES += -DUSE_SOFTOSD
 endif
