diff -NaurwB vdr-1.7.7/epg.c vdr-1.7.7-patched/epg.c
--- vdr-1.7.7/epg.c	2009-06-13 01:01:25.000000000 +0200
+++ vdr-1.7.7-patched/epg.c	2009-11-07 20:40:34.000000000 +0100
@@ -630,6 +630,17 @@
             fprintf(f, "%sX %s\n", Prefix, *p->ToString());
             }
         }
+#ifdef USE_PARENTALRATING
+     if (!isempty(GetContentsString())) {
+       for (int i = 0; i < MAXEVCONTENTS; i++) {
+         if (!isempty(GetContentsString(i))) {
+            strreplace(description, '\n', '|');
+            fprintf(f, "%sG %i %i %s\n",Prefix, Contents(i) & 0xF0, Contents(i) & 0x0F, (const char *)GetContentsString(i));
+            strreplace(description, '|', '\n');
+           }
+         }
+       }
+#endif /* PARENTALRATING */
      if (vps)
         fprintf(f, "%sV %ld\n", Prefix, vps);
      if (!InfoOnly)
@@ -697,6 +708,24 @@
                              }
                           }
                        break;
+#ifdef USE_PARENTALRATING
+             case 'G': if (Event) {
+                          unsigned int ContentID = 0;
+                          unsigned int ContentSubID = 0;
+                          int n = sscanf(t, "%u %u", &ContentID, &ContentSubID);
+                          if (n == 2) {
+                            if (ContentID != 0) {
+                              for (int i = 0; i < MAXEVCONTENTS; i++) {
+                                if (Event->Contents(i) == 0) {
+                                  Event->contents[i] = ContentID | ContentSubID;
+                                  break;
+                                  }
+                                }
+                              }
+                            }
+                          }
+                       break;
+#endif /* PARENTALRATING */
              case 'e': if (Event && !Event->Title())
                           Event->SetTitle(tr("No title"));
                        Event = NULL;
