diff -ruN lzo-2.03.org/Makefile.am lzo-2.03/Makefile.am
--- lzo-2.03.org/Makefile.am	2008-04-30 08:51:49.000000000 -0400
+++ lzo-2.03/Makefile.am	2010-10-29 12:38:51.000000000 -0400
@@ -5,7 +5,7 @@
 
 AUTOMAKE_OPTIONS = 1.9.6 gnu
 
-SUBDIRS = include src lzotest examples tests minilzo
+SUBDIRS = include src
 
 EXTRA_DIST = BUGS
 
diff -ruN lzo-2.03.org/Makefile.in lzo-2.03/Makefile.in
--- lzo-2.03.org/Makefile.in	2008-04-30 08:51:49.000000000 -0400
+++ lzo-2.03/Makefile.in	2010-10-29 12:56:42.000000000 -0400
@@ -193,7 +193,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 AUTOMAKE_OPTIONS = 1.9.6 gnu
-SUBDIRS = include src lzotest examples tests minilzo
+SUBDIRS = include src
 EXTRA_DIST = BUGS
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive

