From 88b4df4530410cd70ab0017a4415e2d040d16c1a Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 01:53:34 -0400
Subject: [PATCH 06/36] added: probe file header for cdxa

---
 libavformat/mpeg.c |   16 ++++++++++++++++
 1 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/libavformat/mpeg.c b/libavformat/mpeg.c
index 135482b..d3b849c 100644
--- a/libavformat/mpeg.c
+++ b/libavformat/mpeg.c
@@ -50,6 +50,18 @@ static int check_pes(uint8_t *p, uint8_t *end){
     return pes1||pes2;
 }
 
+static int cdxa_probe(AVProbeData *p)
+{
+    /* check file header */
+    if (p->buf[0] == 'R' && p->buf[1] == 'I' &&
+        p->buf[2] == 'F' && p->buf[3] == 'F' &&
+        p->buf[8] == 'C' && p->buf[9] == 'D' &&
+        p->buf[10] == 'X' && p->buf[11] == 'A')
+        return AVPROBE_SCORE_MAX;
+    else
+        return 0;
+}
+
 static int mpegps_probe(AVProbeData *p)
 {
     uint32_t code= -1;
@@ -57,6 +69,10 @@ static int mpegps_probe(AVProbeData *p)
     int i;
     int score=0;
 
+    score = cdxa_probe(p);
+    if (score > 0) return score;
+
+    /* Search for MPEG stream */
     for(i=0; i<p->buf_size; i++){
         code = (code<<8) + p->buf[i];
         if ((code & 0xffffff00) == 0x100) {
-- 
1.7.3

