From 065732fbd7649cc90f2e7288f82159cded27cfb2 Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Mon, 28 Jun 2010 02:49:03 -0400
Subject: [PATCH 12/32] mingw workaround

---
 configure |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 3e9c009..60ba27d 100755
--- a/configure
+++ b/configure
@@ -2766,7 +2766,8 @@ check_cflags -Wcast-qual
 check_cflags -Wwrite-strings
 check_cflags -Wtype-limits
 check_cflags -Wundef
-check_cflags -Wmissing-prototypes
+# some gcc compilers (mingw 4.2.1-sjlj) are b0rk so for now skip this
+# check_cflags -Wmissing-prototypes
 enabled extra_warnings && check_cflags -Winline
 
 # add some linker flags
@@ -2825,7 +2826,8 @@ elif enabled ccc; then
 elif enabled gcc; then
     check_cflags -fno-tree-vectorize
     check_cflags -Werror=implicit-function-declaration
-    check_cflags -Werror=missing-prototypes
+# some gcc compilers (mingw 4.2.1-sjlj) are b0rk so for now skip this
+#    check_cflags -Werror=missing-prototypes
 elif enabled clang; then
     check_cflags -Qunused-arguments
 elif enabled armcc; then
-- 
1.7.0

