From 4f4aacd3d85c5caecbc0b0f02739aff0b9d8f657 Mon Sep 17 00:00:00 2001
Date: Mon, 18 Feb 2013 16:58:33 +0000
Subject: [PATCH] allow-customizing-shared-library-soname-name-with-ma

---
 configure |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index 6b9e668..d4aa8f2 100755
--- a/configure
+++ b/configure
@@ -269,6 +269,7 @@ Advanced options (experts only):
   --arch=ARCH              select architecture [$arch]
   --cpu=CPU                select the minimum required CPU (affects
                            instruction selection, may crash on older CPUs)
+  --custom-libname-with-major=NAME  custom library name with major [$SLIBNAME_WITH_MAJOR]
   --enable-pic             build position-independent code
   --enable-sram            allow use of on-chip SRAM
   --enable-thumb           compile for Thumb instruction set
@@ -1527,6 +1528,7 @@ CMDLINE_SET="
     cpu
     cross_prefix
     cxx
+    custom_libname_with_major
     dep_cc
     extra_version
     host_cc
@@ -3408,6 +3410,8 @@ EOF
     exit 1;
 fi
 
+test -n "$custom_libname_with_major" && SLIBNAME_WITH_MAJOR="$custom_libname_with_major"
+
 die_license_disabled() {
     enabled $1 || { enabled $2 && die "$2 is $1 and --enable-$1 is not specified."; }
 }
-- 
1.7.10.4

