From f9cbb94f63aaaf889c21445e71fdd258f9ebcad1 Mon Sep 17 00:00:00 2001
Date: Mon, 18 Feb 2013 17:31:52 +0000
Subject: [PATCH] add-public-version-of-ff_read_frame_flush

---
 libavformat/avformat.h |    5 +++++
 libavformat/utils.c    |    5 +++++
 2 files changed, 10 insertions(+)

diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index c907d4e..c0f517c 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1615,6 +1615,11 @@ int av_read_packet(AVFormatContext *s, AVPacket *pkt);
 int av_read_frame(AVFormatContext *s, AVPacket *pkt);
 
 /**
+ * Clear out any buffered data in context
+ */
+void av_read_frame_flush(AVFormatContext *s);
+
+/**
  * Seek to the keyframe at timestamp.
  * 'timestamp' in 'stream_index'.
  * @param stream_index If stream_index is (-1), a default
diff --git a/libavformat/utils.c b/libavformat/utils.c
index 3e7dec9..72925bb 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -1636,6 +1636,11 @@ void ff_read_frame_flush(AVFormatContext *s)
     }
 }
 
+void av_read_frame_flush(AVFormatContext *s)
+{
+  ff_read_frame_flush(s);
+}
+
 void ff_update_cur_dts(AVFormatContext *s, AVStream *ref_st, int64_t timestamp)
 {
     int i;
-- 
1.7.10.4

