From a356770e3a73d79c9e4a451366930f557e3a2fa6 Mon Sep 17 00:00:00 2001
From: Voyager1 <voyager@xbmc.org>
Date: Sat, 13 Apr 2013 15:38:10 +0200
Subject: [PATCH 4/4] ffmpeg 1.2 - fixed dvd still frames ended up in internal
 lavf

---
 libavformat/utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/utils.c b/libavformat/utils.c
index c3bd3d9..72f59d2 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -706,7 +706,7 @@ no_packet:
 
         if(end || av_log2(pd->buf_size) != av_log2(pd->buf_size - pkt->size)){
             int score= set_codec_from_probe_data(s, st, pd);
-            if(    (st->codec->codec_id != AV_CODEC_ID_NONE && score > AVPROBE_SCORE_RETRY)
+            if(    (st->codec->codec_id != AV_CODEC_ID_NONE && score > AVPROBE_SCORE_RETRY-1)
                 || end){
                 pd->buf_size=0;
                 av_freep(&pd->buf);
-- 
1.8.0.msysgit.0

