From bc85fb8d204c7090dc6856d880f7556e7291d353 Mon Sep 17 00:00:00 2001
Date: Mon, 18 Feb 2013 21:18:39 +0000
Subject: [PATCH] stop-forcing-gas-preprocessor-for-darwin

---
 configure |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index d4aa8f2..980346e 100755
--- a/configure
+++ b/configure
@@ -3166,7 +3166,7 @@ case $target_os in
         strip="strip -d"
         ;;
     darwin)
-        gas="gas-preprocessor.pl $cc"
+        #gas="gas-preprocessor.pl $cc"
         enabled ppc && add_asflags -force_cpusubtype_ALL
         SHFLAGS='-dynamiclib -Wl,-single_module -Wl,-install_name,$(SHLIBDIR)/$(SLIBNAME_WITH_MAJOR),-current_version,$(LIBVERSION),-compatibility_version,$(LIBMAJOR)'
         enabled x86_32 && append SHFLAGS -Wl,-read_only_relocs,suppress
-- 
1.7.10.4

