Index: libavformat/matroskadec.c
===================================================================
--- libavformat/matroskadec.c	(revision 24796)
+++ libavformat/matroskadec.c	(working copy)
@@ -1435,6 +1435,10 @@
             st->codec->channels = track->audio.channels;
             if (st->codec->codec_id != CODEC_ID_AAC)
             st->need_parsing = AVSTREAM_PARSE_HEADERS;
+            if (st->codec->codec_id == CODEC_ID_DTS) {
+                st->need_parsing = AVSTREAM_PARSE_FULL;
+                st->codec->channels = 0;
+            }
         } else if (track->type == MATROSKA_TRACK_TYPE_SUBTITLE) {
             st->codec->codec_type = AVMEDIA_TYPE_SUBTITLE;
         }
