diff --git a/lib/ffmpeg/libavformat/rtsp.h b/lib/ffmpeg/libavformat/rtsp.h
index f67ceb8..fa607bb 100644
--- a/lib/ffmpeg/libavformat/rtsp.h
+++ b/lib/ffmpeg/libavformat/rtsp.h
@@ -35,8 +35,8 @@
  * Network layer over which RTP/etc packet data will be transported.
  */
 enum RTSPLowerTransport {
-    RTSP_LOWER_TRANSPORT_UDP = 0,           /**< UDP/unicast */
-    RTSP_LOWER_TRANSPORT_TCP = 1,           /**< TCP; interleaved in RTSP */
+    RTSP_LOWER_TRANSPORT_TCP = 0,           /**< TCP; interleaved in RTSP */
+    RTSP_LOWER_TRANSPORT_UDP = 1,           /**< UDP/unicast */
     RTSP_LOWER_TRANSPORT_UDP_MULTICAST = 2, /**< UDP/multicast */
     RTSP_LOWER_TRANSPORT_NB,
     RTSP_LOWER_TRANSPORT_HTTP = 8,          /**< HTTP tunneled - not a proper

