From 5a25200ef7dca766d66e35df35adf4fc519a6229 Mon Sep 17 00:00:00 2001
From: Anssi Hannula <anssi.hannula@iki.fi>
Date: Fri, 21 Jan 2011 20:44:02 +0200
Subject: [PATCH 44/46] aacenc: add recognized profiles array

Submitted upstream.

---
 libavcodec/aacenc.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/libavcodec/aacenc.c b/libavcodec/aacenc.c
index c52ffa0..762e0c8 100644
--- a/libavcodec/aacenc.c
+++ b/libavcodec/aacenc.c
@@ -636,6 +636,11 @@ static av_cold int aac_encode_end(AVCodecContext *avctx)
     return 0;
 }
 
+static const AVProfile profiles[] = {
+    { FF_PROFILE_AAC_LOW,  "Low"  },
+    { FF_PROFILE_UNKNOWN },
+};
+
 AVCodec aac_encoder = {
     "aac",
     AVMEDIA_TYPE_AUDIO,
@@ -647,4 +652,5 @@ AVCodec aac_encoder = {
     .capabilities = CODEC_CAP_SMALL_LAST_FRAME | CODEC_CAP_DELAY | CODEC_CAP_EXPERIMENTAL,
     .sample_fmts = (const enum AVSampleFormat[]){AV_SAMPLE_FMT_S16,AV_SAMPLE_FMT_NONE},
     .long_name = NULL_IF_CONFIG_SMALL("Advanced Audio Coding"),
+    .profiles = profiles,
 };
-- 
1.7.3

