From dc0509f7a6341a72ca4399fee37e30c91fddc222 Mon Sep 17 00:00:00 2001
From: arnova <nospam@void.org>
Date: Tue, 8 Nov 2011 08:35:59 +0100
Subject: [PATCH] fixed: Typo

---
 lib/ffmpeg/libavformat/mmsh.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/ffmpeg/libavformat/mmsh.c b/lib/ffmpeg/libavformat/mmsh.c
index 3ae886f..cc6efa3 100644
--- a/lib/ffmpeg/libavformat/mmsh.c
+++ b/lib/ffmpeg/libavformat/mmsh.c
@@ -284,7 +284,7 @@ static int mmsh_open_internal(URLContext *h, const char *uri, int flags, int tim
                    CLIENTGUID
                    "Pragma: stream-switch-count=%d\r\n"
                    "Pragma: stream-switch-entry=%s\r\n"
-                   "Pragma: no-cache,rate)1.000000,stream-time=%u"
+                   "Pragma: no-cache,rate=1.000000,stream-time=%u"
                    "Connection: Close\r\n\r\n",
                    host, port, mmsh->request_seq++, mms->stream_num, stream_selection, timestamp);
     av_freep(&stream_selection);
-- 
1.7.7.2

