--- a/lib/fribidi-common.h
+++ b/lib/fribidi-common.h
@@ -43,17 +43,7 @@
 
 /* FRIBIDI_ENTRY is a macro used to declare library entry points. */
 #ifndef FRIBIDI_ENTRY
-# if (defined(_MSC_VER) || defined(FRIBIDI_BUILT_WITH_MSVC)) && !defined(FRIBIDI_STATIC)
-/* if we're building fribidi itself with MSVC, FRIBIDI_ENTRY will be defined,
- * so if we're here then this is an external user including fribidi headers.
- * The dllimport is needed here mostly for the fribidi_version_info variable,
- * for functions it's not required. Probably needs more fine-tuning if
- * someone starts building fribidi as static library with MSVC. We'll cross
- * that bridge when we get there. */
-#  define FRIBIDI_ENTRY __declspec(dllimport) extern
-# else
 #  define FRIBIDI_ENTRY extern
-# endif
 #endif /* !FRIBIDI_ENTRY */
 
 #ifdef __ICC
