--- a/src/dvd_input.h
+++ b/src/dvd_input.h
@@ -35,6 +35,13 @@ typedef struct dvd_input_s *dvd_input_t;
 #if defined( __MINGW32__ )
 #   undef  lseek
 #   define lseek  _lseeki64
+#   undef  fseeko
+#   define fseeko fseeko64
+#   undef  ftello
+#   define ftello ftello64
+#   define flockfile(...)
+#   define funlockfile(...)
+#   define getc_unlocked getc
 #   undef  off_t
 #   define off_t off64_t
 #   undef  stat
