--- src/Makefile.am.old	2011-09-23 23:14:39.000000000 +0200
+++ src/Makefile.am	2011-09-25 21:38:08.000000000 +0200
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES=libshairport.la
 
-library_includedir=$(includedir)
-library_include_HEADERS = shairport.h
+library_includedir=$(includedir)/shairport
+library_include_HEADERS = shairport.h ao.h socketlib.h
 
 libshairport_la_SOURCES=shairport.c hairtunes.c socketlib.c alac.c
 libshairport_la_LDFLAGS=-dynamiclib
