Index: include/SDL_video.h
===================================================================
--- include/SDL_video.h	(revision 4116)
+++ include/SDL_video.h	(working copy)
@@ -324,6 +324,11 @@
 extern DECLSPEC SDL_Rect ** SDLCALL SDL_ListModes(SDL_PixelFormat *format, Uint32 flags);
 
 /**
+* Alter the width and height of the current surface to the given sizes.
+*/
+extern DECLSPEC void SDLCALL SDL_SetWidthHeight(int width, int height);
+
+/**
  * Set up a video mode with the specified width, height and bits-per-pixel.
  *
  * If 'bpp' is 0, it is treated as the current display bits per pixel.

Index: src/video/SDL_video.c
===================================================================
--- src/video/SDL_video.c	(revision 4116)
+++ src/video/SDL_video.c	(working copy)
@@ -1956,3 +1956,11 @@
 		return(0);
 	}
 }
+
+void SDL_SetWidthHeight(int width, int height)
+{
+  if (current_video != NULL && current_video->screen != NULL) {
+    current_video->screen->w = width;
+    current_video->screen->h = height;
+  }
+}
