--- a/src/video/quartz/SDL_QuartzEvents.m
+++ b/src/video/quartz/SDL_QuartzEvents.m
@@ -345,7 +345,11 @@
         the scancode/keysym.
     */
     if (SDL_TranslateUNICODE && state == SDL_PRESSED) {
-        [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
+        NSResponder *firstResponder = [[NSApp keyWindow] firstResponder];
+        if ([NSStringFromClass([firstResponder class]) isEqual:@"OSXTextInputResponder"])
+          [firstResponder interpretKeyEvents:[NSArray arrayWithObject:event]];
+        else
+          [field_edit interpretKeyEvents:[NSArray arrayWithObject:event]];
         chars = [ event characters ];
         numChars = [ chars length ];
         if (numChars > 0)
