--- src/video/quartz/SDL_QuartzVideo.h	2012-01-19 07:30:06.000000000 +0100
+++ src/video/quartz/SDL_QuartzVideo.h	2013-10-30 16:35:02.000000000 +0100
@@ -91,7 +91,9 @@
     CGDirectDisplayID  display;            /* 0 == main display (only support single display) */
     const void         *mode;              /* current mode of the display */
     const void         *save_mode;         /* original mode of the display */
+#ifdef CGDirectPaletteRef
     CGDirectPaletteRef palette;            /* palette of an 8-bit display */
+#endif
     NSOpenGLContext    *gl_context;        /* OpenGL rendering context */
     NSGraphicsContext  *nsgfx_context;     /* Cocoa graphics context */
     Uint32             width, height, bpp; /* frequently used data about the display */
