From 763a3322e981e0493106b87ae1c86c0054f0da2d Mon Sep 17 00:00:00 2001
From: "h.udo" <hudokkow@gmail.com>
Date: Thu, 20 Oct 2016 15:22:30 +0100
Subject: [PATCH 10/12] [configure] Fix build with xcode5

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3ada0ae..d9a1771 100644
--- a/configure.ac
+++ b/configure.ac
@@ -119,7 +119,7 @@ if test "$enable_threads" != no; then
       [AC_LINK_IFELSE(
 [AC_LANG_SOURCE([#include <pthread.h>
 
-int main(int argc, char *argv[]) {
+int main(int argc, char **argv) {
 	pthread_mutex_t mutex;
 	
 	pthread_mutex_init(&mutex, NULL);
@@ -138,7 +138,7 @@ int main(int argc, char *argv[]) {
       [AC_LINK_IFELSE(
 [AC_LANG_SOURCE([#include <windows.h>
 
-int main(int argc, char *argv[]) {
+int main(int argc, char **argv) {
   CreateMutex(NULL, FALSE, NULL);
   return 0;
 }
-- 
2.14.1

