diff -ruN configure.in configure.in
--- configure.in	2007-09-13 11:48:42.000000000 -0400
+++ configure.in	2012-06-10 20:13:50.543973093 -0400
@@ -269,10 +269,6 @@
 # only matters for x86
 AC_CHECK_PROGS(NASM, nasm)
 AM_CONDITIONAL(FLaC__HAS_NASM, test -n "$NASM")
-if test -n "$NASM" ; then
-AC_DEFINE(FLAC__HAS_NASM)
-AH_TEMPLATE(FLAC__HAS_NASM, [define if you are compiling for x86 and have the NASM assembler])
-fi
 
 # only matters for PowerPC
 AC_CHECK_PROGS(AS, as, as)
