diff -ruN libmad-0.15.1b.orig/imdct_l_arm.S libmad-0.15.1b/imdct_l_arm.S
--- libmad-0.15.1b.orig/imdct_l_arm.S.orig	2010-10-28 18:57:51.000000000 -0400
+++ libmad-0.15.1b/imdct_l_arm.S	2010-10-28 18:58:41.000000000 -0400
@@ -468,8 +468,11 @@
 
     @----
 
-    add     r2, pc, #(imdct36_long_karray-.-8)  @ r2 = base address of Knn array (PIC safe ?)
-
+#ifdef __thumb__
+    adr     r2, imdct36_long_karray
+#else
+     add     r2, pc, #(imdct36_long_karray-.-8)  @ r2 = base address of Knn array (PIC safe ?)
+#endif
 
 loop:
     ldr     r12, [r0, #X0]
