--- libdvdnav-4.2.0/src/dvdnav/dvdnav.h	Wed Mar 20 11:38:52 2013
+++ lib/libdvd/libdvdnav/src/dvdnav/dvdnav.h	Wed Mar 20 13:19:13 2013
@@ -63,12 +63,6 @@
 #define DVDNAV_STATUS_ERR 0
 #define DVDNAV_STATUS_OK  1
 
-#define DVDNAV_FORMAT_AC3 0
-#define DVDNAV_FORMAT_MPEGAUDIO 3
-#define DVDNAV_FORMAT_LPCM 4
-#define DVDNAV_FORMAT_DTS 5
-#define DVDNAV_FORMAT_SDDS 6
-
 /*********************************************************************
  * initialisation & housekeeping functions                           *
  *********************************************************************/
