--- libdvdnav-4.2.0/src/dvdnav/dvd_types.h	Wed Mar 20 08:51:10 2013
+++ lib/libdvd/libdvdnav/src/dvdnav/dvd_types.h	Wed Mar 20 12:22:25 2013
@@ -161,14 +161,13 @@
 /* The audio format */
 typedef enum {
   DVD_AUDIO_FORMAT_AC3       = 0,
-  DVD_AUDIO_FORMAT_MPEG1     = 1,
-  DVD_AUDIO_FORMAT_MPEG1_DRC = 2,
-  DVD_AUDIO_FORMAT_MPEG2     = 3,
-  DVD_AUDIO_FORMAT_MPEG2_DRC = 4,
-  DVD_AUDIO_FORMAT_LPCM      = 5,
+  DVD_AUDIO_FORMAT_UNKNOWN_1 = 1,
+  DVD_AUDIO_FORMAT_MPEG      = 2,
+  DVD_AUDIO_FORMAT_MPEG2_EXT = 3,
+  DVD_AUDIO_FORMAT_LPCM      = 4,
+  DVD_AUDIO_FORMAT_UNKNOWN_5 = 5,
   DVD_AUDIO_FORMAT_DTS       = 6,
-  DVD_AUDIO_FORMAT_SDDS      = 7,
-  DVD_AUDIO_FORMAT_Other     = 8
+  DVD_AUDIO_FORMAT_SDDS      = 7
 } DVDAudioFormat_t;
 
 /* Audio language extension */
