--- mpi/longlong.h	2005-07-29 09:31:17.000000000 -0400
+++ mpi/longlong.h	2013-01-20 15:18:22.000000000 -0500
@@ -223,15 +223,24 @@
 	   : "r" ((USItype)(a)),                                        \
 	     "r" ((USItype)(b))                                         \
 	   : "r0", "r1", "r2")
-#else
+#elif defined(__APPLE__)
 #define umul_ppmm(xh, xl, a, b)                                         \
-  __asm__ ("%@ Inlined umul_ppmm\n"                                     \
-	   "umull %r1, %r0, %r2, %r3"                                   \
+  __asm__ ("@ Inlined umul_ppmm\n"                                      \
+	   "umull %1, %0, %2, %3"                                       \
 		   : "=&r" ((USItype)(xh)),                             \
 		     "=r" ((USItype)(xl))                               \
 		   : "r" ((USItype)(a)),                                \
 		     "r" ((USItype)(b))                                 \
 		   : "r0", "r1")
+#else
+#define umul_ppmm(xh, xl, a, b)                                         \
+  __asm__ ("%@ Inlined umul_ppmm\n"                                     \
+          "umull %r1, %r0, %r2, %r3"                                    \
+                   : "=&r" ((USItype)(xh)),                             \
+                     "=r" ((USItype)(xl))                               \
+                   : "r" ((USItype)(a)),                                \
+                     "r" ((USItype)(b))                                 \
+		   : "r0", "r1")
 #endif
 #define UMUL_TIME 20
 #define UDIV_TIME 100
