--- cipher/Makefile.am.orig	2013-03-23 23:55:30.000000000 +0100
+++ cipher/Makefile.am	2013-03-24 00:03:34.000000000 +0100
@@ -68,7 +68,7 @@
 camellia.c camellia.h camellia-glue.c
 
 if ENABLE_O_FLAG_MUNGING
-o_flag_munging = sed -e 's/-O[2-9s]*/-O1/g'
+o_flag_munging = sed -e 's/ -O[2-9s]* / -O1 /g'
 else
 o_flag_munging = cat
 endif
--- cipher/Makefile.in.orig	2013-03-24 00:03:10.000000000 +0100
+++ cipher/Makefile.in	2013-03-24 00:03:25.000000000 +0100
@@ -274,7 +274,7 @@
 camellia.c camellia.h camellia-glue.c
 
 @ENABLE_O_FLAG_MUNGING_FALSE@o_flag_munging = cat
-@ENABLE_O_FLAG_MUNGING_TRUE@o_flag_munging = sed -e 's/-O[2-9s]*/-O1/g'
+@ENABLE_O_FLAG_MUNGING_TRUE@o_flag_munging = sed -e 's/ -O[2-9s]* / -O1 /g'
 all: all-am
 
 .SUFFIXES:
