diff -rup pdk-org/default_config/defaults.cfg pdk/default_config/defaults.cfg
--- pdk-org/default_config/defaults.cfg	2008-05-31 10:02:23.000000000 -0400
+++ pdk/default_config/defaults.cfg	2008-05-31 10:09:18.000000000 -0400
@@ -117,6 +117,12 @@ buildroot_extras = squashfs-tools dosfst
 # What packages need to be installed in the buildroot (outside the rootstrap archive)
 buildroot_packages = grub man less vim automake autotools-dev openssh-clients
 
+# ***************** XBMC Live CD/USB Platform ********************
+[platform.xbmc-ubuntu-hardy]
+buildroot_mirror =  http://archive.ubuntu.com/ubuntu/
+buildroot_codename = hardy
+architecture=i386
+
 # ********** Install Image section ***********
 [installimage]
 welcome_message = Welcome to Moblin:
@@ -170,3 +176,4 @@ required_exes = debootstrap apt-get
 [distribution.fedora]
 # It is blank, but don't delete it
 required_exes =
+
diff -rup pdk-org/lib/InstallImage.py pdk/lib/InstallImage.py
--- pdk-org/lib/InstallImage.py	2008-05-31 10:02:23.000000000 -0400
+++ pdk/lib/InstallImage.py	2008-05-31 10:05:51.000000000 -0400
@@ -248,7 +248,7 @@ class InstallImage(object):
         fs_path      = self.target.fs_path[len(self.project.path):]
         image_path   = self.target.image_path[len(self.project.path):]
         image_path   = os.path.join(image_path,'rootfs.img')
-        cmd          = "mksquashfs %s %s -no-progress -ef %s" % (fs_path, image_path, self.exclude_file)
+        cmd          = "mksquashfs %s %s -no-sparse -no-progress -ef %s" % (fs_path, image_path, self.exclude_file)
         self.write_manifest(self.path)
         self.target.umount()
         print "Executing the mksquashfs program: %s" % cmd
@@ -282,7 +282,7 @@ class InstallImage(object):
         fs_path    = os.path.join(fs_path, 'boot')
         image_path = self.target.image_path[len(self.project.path):]
         image_path = os.path.join(image_path,'bootfs.img')
-        cmd        = "mksquashfs %s %s -no-progress" % (fs_path, image_path)
+        cmd        = "mksquashfs %s %s -no-sparse -no-progress" % (fs_path, image_path)
         self.project.chroot(cmd)
 
     def delete_bootfs(self):
diff -rup pdk-org/platforms/platforms.cfg pdk/platforms/platforms.cfg
--- pdk-org/platforms/platforms.cfg	2008-05-31 10:02:23.000000000 -0400
+++ pdk/platforms/platforms.cfg	2008-05-31 10:05:51.000000000 -0400
@@ -42,3 +42,9 @@ target_os = ubuntu
 description = [BETA] McCaslin on LPIA architecture (e.g. Samsung Q1 Ultra) using Fedora
 package_manager = yum
 target_os = fedora
+
+[xbmc-ubuntu-hardy]
+description = XBMC for Linux on Generic i386 Platform using Hardy
+package_manager = apt
+target_os = ubuntu
+
