--- a/lib/driver/MSWindows/win32_ioctl.c
+++ b/lib/driver/MSWindows/win32_ioctl.c
@@ -83,10 +83,10 @@
                         error_msg, sizeof(error_msg), NULL);             \
   (count != 0) ?                                                         \
     cdio_log(loglevel, "Error: file %s: line %d (%s)\n\t%s\n",           \
-             __FILE__, __LINE__, __PRETTY_FUNCTION__, error_msg)         \
+             __FILE__, __LINE__, __FUNCTION__, error_msg)         \
     :                                                                    \
     cdio_log(loglevel, "Error: file %s: line %d (%s) %ld\n",             \
-             __FILE__, __LINE__, __PRETTY_FUNCTION__, (long int) i_err); \
+             __FILE__, __LINE__, __FUNCTION__, (long int) i_err); \
 }
 #endif
 
