--- encode.c	2009-11-02 06:46:30.000000000 -0500
+++ encode.c	2013-03-13 19:47:36.156302056 -0400
@@ -426,7 +426,7 @@
 
 #ifdef HAVE_LIBZ
 
-#include "Zip.h"
+#include "libImaging/Zip.h"
 
 PyObject*
 PyImaging_ZipEncoderNew(PyObject* self, PyObject* args)
--- decode.c	2009-11-02 06:46:46.000000000 -0500
+++ decode.c	2013-03-13 19:47:53.326302045 -0400
@@ -602,7 +602,7 @@
 
 #ifdef HAVE_LIBZ
 
-#include "Zip.h"
+#include "libImaging/Zip.h"
 
 PyObject*
 PyImaging_ZipDecoderNew(PyObject* self, PyObject* args)
