--- a/setup.py
+++ b/setup.py
@@ -414,6 +414,8 @@
         if _cmd_exists(os.environ.get("PKG_CONFIG", "pkg-config")):
             pkg_config = _pkg_config
 
+        self.compiler.include_dirs = [os.environ["PYTHONXINCLUDE"]]
+
         #
         # add configured kits
         for root_name, lib_name in dict(
@@ -471,8 +473,8 @@
                 for d in os.environ[k].split(os.path.pathsep):
                     _add_directory(library_dirs, d)
 
-        _add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
-        _add_directory(include_dirs, os.path.join(sys.prefix, "include"))
+        #_add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
+        #_add_directory(include_dirs, os.path.join(sys.prefix, "include"))
 
         #
         # add platform directories
