--- a/setup.py
+++ b/setup.py
@@ -363,6 +363,16 @@
         if _cmd_exists(os.environ.get("PKG_CONFIG", "pkg-config")):
             pkg_config = _pkg_config
 
+        compiler = os.environ.get('CC')
+        args = {}
+        # unfortunately, distutils doesn't let us provide separate C and C++
+        # compilers
+        if compiler is not None:
+            ccshared = os.environ.get('CCSHARED') or ''
+            cflags = os.environ.get('CFLAGS') or ''
+            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
+        self.compiler.set_executables(**args)
+
         #
         # add configured kits
         for root_name, lib_name in dict(
@@ -418,10 +428,11 @@
                 for d in os.environ[k].split(os.path.pathsep):
                     _add_directory(library_dirs, d)
 
-        prefix = sysconfig.get_config_var("prefix")
+        prefix = os.environ.get('PYTHONXCPREFIX')
         if prefix:
             _add_directory(library_dirs, os.path.join(prefix, "lib"))
             _add_directory(include_dirs, os.path.join(prefix, "include"))
+            _add_directory(include_dirs, os.path.join(prefix, "include/python3.8"))
 
         #
         # add platform directories
