--- a/setup.py
+++ b/setup.py
@@ -477,6 +477,8 @@ class pil_build_ext(build_ext):
         if _cmd_exists(os.environ.get("PKG_CONFIG", "pkg-config")):
             pkg_config = _pkg_config
 
+        self.compiler.include_dirs = [os.environ["PYTHONXINCLUDE"]]
+
         #
         # add configured kits
         for root_name, lib_name in {
@@ -547,8 +549,8 @@ class pil_build_ext(build_ext):
                 for d in os.environ[k].split(os.path.pathsep):
                     _add_directory(library_dirs, d)
 
-        _add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
-        _add_directory(include_dirs, os.path.join(sys.prefix, "include"))
+        #_add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
+        #_add_directory(include_dirs, os.path.join(sys.prefix, "include"))
 
         #
         # add platform directories
