--- a/setup.py
+++ b/setup.py
@@ -414,6 +414,19 @@
         if _cmd_exists(os.environ.get("PKG_CONFIG", "pkg-config")):
             pkg_config = _pkg_config
 
+        compiler = os.environ.get('CC')
+        args = {}
+        # unfortunately, distutils doesn't let us provide separate C and C++
+        # compilers
+        if compiler is not None:
+            ccshared = os.environ.get('CCSHARED') or ''
+            cflags = os.environ.get('CFLAGS') or ''
+            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
+        self.compiler.set_executables(**args)
+
+        self.compiler.include_dirs.clear()
+        self.compiler.include_dirs.append(os.environ.get('PYTHONXINCLUDE'))
+
         #
         # add configured kits
         for root_name, lib_name in dict(
