From 8d59051446b38b9c73a7eae7359341483c6951f7 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sat, 10 Apr 2021 08:16:48 -0700
Subject: [PATCH 2/4] ffmpeg: windows: configure: fix zlib conflict

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index c4fe14568a..9bebb75093 100755
--- a/configure
+++ b/configure
@@ -8234,6 +8234,9 @@ print_config HAVE_   "$config_files" $HAVE_LIST
 print_config CONFIG_ "$config_files" $CONFIG_LIST       \
                                      $CONFIG_EXTRA      \
 
+echo "#if defined(HAVE_UNISTD_H) && HAVE_UNISTD_H == 0" >> $TMPH
+echo "#undef HAVE_UNISTD_H" >> $TMPH
+echo "#endif" >> $TMPH
 echo "#endif /* FFMPEG_CONFIG_H */" >> $TMPH
 
 # Do not overwrite an unchanged config.h to avoid superfluous rebuilds.
-- 
2.46.1

