--- a/src/spawn-posix.c
+++ b/src/spawn-posix.c
@@ -314,7 +314,7 @@
 
   if (preexec)
     preexec ();
-  execv (pgmname, arg_list);
+//  execv (pgmname, arg_list);
   /* No way to print anything, as we have may have closed all streams. */
   _exit (127);
 }
@@ -464,7 +464,7 @@
     }
 
   _gpgrt_pre_syscall ();
-  *pid = fork ();
+  *pid = -1;
   _gpgrt_post_syscall ();
   if (*pid == (pid_t)(-1))
     {
@@ -534,7 +534,7 @@
   gpg_error_t err;
 
   _gpgrt_pre_syscall ();
-  *pid = fork ();
+  *pid = -1;
   _gpgrt_post_syscall ();
   if (*pid == (pid_t)(-1))
     {
@@ -813,7 +813,7 @@
     return _gpg_err_code_from_syserror ();
 
   _gpgrt_pre_syscall ();
-  pid = fork ();
+  pid = -1;
   _gpgrt_post_syscall ();
   if (pid == (pid_t)(-1))
     {
@@ -830,7 +830,7 @@
       if (setsid() == -1 || chdir ("/"))
         _exit (1);
 
-      pid2 = fork (); /* Double fork to let init take over the new child. */
+      pid2 = -1; /* Double fork to let init take over the new child. */
       if (pid2 == (pid_t)(-1))
         _exit (1);
       if (pid2)
