--- afpfs-ng-0.8.1/lib/Makefile.am	2008-02-18 04:34:32.000000000 +0100
+++ afpfs-ng-0.8.1/lib/Makefile.am	2011-08-26 21:33:33.000000000 +0200
@@ -4,7 +4,7 @@
 
 lib_LTLIBRARIES = libafpclient.la
 
-libafpclient_la_SOURCES = afp.c codepage.c did.c dsi.c map_def.c uams.c uams_def.c unicode.c users.c utils.c resource.c log.c client.c server.c connect.c loop.c midlevel.c proto_attr.c proto_desktop.c proto_directory.c proto_files.c proto_fork.c proto_login.c proto_map.c proto_replyblock.c proto_server.c proto_volume.c proto_session.c afp_url.c status.c forklist.c debug.c lowlevel.c
+libafpclient_la_SOURCES = afp.c codepage.c did.c dsi.c map_def.c uams.c uams_def.c unicode.c users.c utils.c resource.c log.c client.c server.c connect.c loop.c midlevel.c proto_attr.c proto_desktop.c proto_directory.c proto_files.c proto_fork.c proto_login.c proto_map.c proto_replyblock.c proto_server.c proto_volume.c proto_session.c afp_url.c status.c forklist.c debug.c lowlevel.c identify.c
 
 # libafpclient_la_LDFLAGS = -module -avoid-version
 
