--- a/builds/cmake/iOS.cmake
+++ b/builds/cmake/iOS.cmake
@@ -80,7 +80,7 @@ set(CMAKE_OSX_DEPLOYMENT_TARGET ""
 # SDKs
 find_program(CMAKE_UNAME uname /bin /usr/bin /usr/local/bin)
 if (CMAKE_UNAME)
-  exec_program(uname ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
+  execute_process(COMMAND uname -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
   string(REGEX REPLACE "^([0-9]+)\\.([0-9]+).*$" "\\1"
     DARWIN_MAJOR_VERSION "${CMAKE_HOST_SYSTEM_VERSION}")
 endif (CMAKE_UNAME)
