--- config/Makefile.darwin.orig	2010-11-04 07:55:35.000000000 +1100
+++ config/Makefile.darwin	2010-11-16 18:28:44.000000000 +1100
@@ -1,13 +1,13 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
-CFLAGS.EXTRA = -no-cpp-precomp -dynamic -fno-common
+CC = $(CC.ARCH)
+LD = $(CC.ARCH)
+CFLAGS.EXTRA = -dynamic -fno-common $(CFLAGS.ARCH)
 ifneq (undefined, $(origin GLEW_MX))
 CFLAGS.EXTRA += -DGLEW_MX
 endif
 PICFLAG = -fPIC
 LDFLAGS.SO = -dynamiclib -install_name $(GLEW_DEST)/lib/$(LIB.SHARED)
-LDFLAGS.EXTRA =
+LDFLAGS.EXTRA = $(LDFLAGS.ARCH)
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
 LDFLAGS.GL = -L/usr/X11R6/lib -lXmu -lXi -lGLU -lGL -lXext -lX11
