diff -ruN libvorbis-1.2.0/configure libvorbis-1.2.0.mod/configure
--- libvorbis-1.2.0/configure	2007-07-25 12:46:37.000000000 -0400
+++ libvorbis-1.2.0.mod/configure	2010-10-29 11:46:06.000000000 -0400
@@ -19476,9 +19476,9 @@
 		CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8"
 		PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
 	*-*-darwin*)
-		DEBUG="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -O0 -fsigned-char"
-		CFLAGS="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -O4 -ffast-math -fsigned-char"
-		PROFILE="-DDARWIN -fno-common -force_cpusubtype_ALL -Wall -g -pg -O4 -ffast-math -fsigned-char";;
+		DEBUG="-DARWIN_NO_CARBON -fno-common -noforce_cpusubtype_ALL -Wall -g -O0 -fsigned-char"
+		CFLAGS="-DARWIN_NO_CARBON -fno-common -noforce_cpusubtype_ALL -Wall -g -O4 -ffast-math -fsigned-char"
+		PROFILE="-DARWIN_NO_CARBON -fno-common -noforce_cpusubtype_ALL -Wall -g -pg -O4 -ffast-math -fsigned-char";;
 	*)
 		DEBUG="-g -Wall -W -D__NO_MATH_INLINES -fsigned-char"
 		CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char"

