--- a/lib/accelerated/aarch64/Makefile.am
+++ b/lib/accelerated/aarch64/Makefile.am
@@ -34,7 +34,7 @@
 endif
 
 #ensure that we have all aarch64 instruction sets enabled for the assembler
-AM_CCASFLAGS = -Wa,-march=all
+#AM_CCASFLAGS = -Wa,-march=all
 
 EXTRA_DIST = README
 
--- a/lib/accelerated/aarch64/Makefile.in
+++ b/lib/accelerated/aarch64/Makefile.in
@@ -1634,7 +1634,7 @@
 	-I$(srcdir)/../../ -I$(srcdir)/../ $(am__append_1)
 
 #ensure that we have all aarch64 instruction sets enabled for the assembler
-AM_CCASFLAGS = -Wa,-march=all
+#AM_CCASFLAGS = -Wa,-march=all
 EXTRA_DIST = README
 noinst_LTLIBRARIES = libaarch64.la
 libaarch64_la_SOURCES = aarch64-common.c aarch64-common.h \
