diff -ruN src/daemon/daemon.c src/daemon/daemon.c
--- src/daemon/daemon.c	2012-05-07 17:21:43.695832026 -0400
+++ src/daemon/daemon.c	2012-05-07 17:20:20.365832079 -0400
@@ -244,7 +244,7 @@
   MHD_ip_count_lock (daemon);
 
   /* Search for the IP address */
-  node = (void*)TSEARCH (key, &daemon->per_ip_connection_count, MHD_ip_addr_compare);
+//  node = (void*)TSEARCH (key, &daemon->per_ip_connection_count, MHD_ip_addr_compare);
   if (!node)
     {
 #if HAVE_MESSAGES
@@ -296,7 +296,7 @@
   MHD_ip_count_lock (daemon);
 
   /* Search for the IP address */
-  node = (void*)TFIND (&search_key, &daemon->per_ip_connection_count, MHD_ip_addr_compare);
+//  node = (void*)TFIND (&search_key, &daemon->per_ip_connection_count, MHD_ip_addr_compare);
 
   /* Something's wrong if we couldn't find an IP address
    * that was previously added */
diff -ruN src/include/platform.h src/include/platform.h
--- src/include/platform.h	2012-05-07 17:34:57.345831526 -0400
+++ src/include/platform.h	2012-05-07 17:20:20.375832079 -0400
@@ -53,7 +53,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <signal.h>
-#if !defined(MINGW) && !defined(__SYMBIAN32__)
+#if !defined(MINGW) && !defined(__SYMBIAN32__) && !defined(ANDROID)
 #include <search.h>
 #endif
 #include <stddef.h>
