--- a/lib/driver/image/bincue.c
+++ b/lib/driver/image/bincue.c
@@ -1,3 +1,4 @@
+
 /*
   Copyright (C) 2002-2006, 2008, 2011-2012, 2014, 2017
     Rocky Bernstein <rocky@gnu.org>
@@ -1045,7 +1046,7 @@
 {
   char **drives = NULL;
   unsigned int num_files=0;
-#ifdef HAVE_GLOB_H
+#if defined(HAVE_GLOB_H) && !defined(__ANDROID__)
   unsigned int i;
   glob_t globbuf;
   globbuf.gl_offs = 0;
--- a/lib/driver/image/cdrdao.c
+++ b/lib/driver/image/cdrdao.c
@@ -1149,7 +1149,7 @@
 {
   char **drives = NULL;
   unsigned int num_files=0;
-#ifdef HAVE_GLOB_H
+#if defined(HAVE_GLOB_H) && !defined(__ANDROID__)
   unsigned int i;
   glob_t globbuf;
   globbuf.gl_offs = 0;
--- a/lib/driver/image/nrg.c
+++ b/lib/driver/image/nrg.c
@@ -1167,7 +1167,7 @@
 {
   char **drives = NULL;
   unsigned int num_files=0;
-#ifdef HAVE_GLOB_H
+#if defined(HAVE_GLOB_H) && !defined(__ANDROID__)
   unsigned int i;
   glob_t globbuf;
   globbuf.gl_offs = 0;
