From 8e6795779558d2828aef14078fefbcadd5323fa6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juho=20V=C3=A4h=C3=A4-Herttua?= <juhovh@iki.fi>
Date: Mon, 24 Mar 2014 21:43:59 +0200
Subject: [PATCH] Make password authentication work on iTunes again, fixes #20.

---
 src/lib/raop.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/lib/raop.c b/src/lib/raop.c
index e5c6539..a58e92f 100644
--- a/src/lib/raop.c.orig	2013-04-17 15:17:49.000000000 +0200
+++ b/src/lib/raop.c	2014-03-24 23:22:30.000000000 +0100
@@ -139,7 +139,7 @@
 	}
 
 	res = http_response_init("RTSP/1.0", 200, "OK");
-	if (strlen(raop->password)) {
+	if (strcmp(method, "OPTIONS") && strlen(raop->password)) {
 		const char *authorization;
 
 		authorization = http_request_get_header(request, "Authorization");
