--- a/include/wayland-client.hpp
+++ b/include/wayland-client.hpp
@@ -33,6 +33,7 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <cstdint>
 #include <wayland-version.hpp>
 #include <wayland-client-core.h>
 #include <wayland-util.hpp>
--- a/include/wayland-server.hpp
+++ b/include/wayland-server.hpp
@@ -27,6 +27,7 @@
 #define WAYLAND_SERVER_HPP
 
 #include <atomic>
+#include <cstdint>
 #include <functional>
 #include <list>
 #include <memory>
--- a/scanner/scanner.cpp
+++ b/scanner/scanner.cpp
@@ -24,6 +24,7 @@
 #include <cctype>
 #include <cmath>
 #include <stdexcept>
+#include <cstdint>
 
 #include "pugixml.hpp"
 
@@ -1110,6 +1111,7 @@ int main(int argc, char *argv[])
               << "#include <memory>" << std::endl
               << "#include <string>" << std::endl
               << "#include <vector>" << std::endl
+              << "#include <cstdint>" << std::endl
               << std::endl
               << (server ? "#include <wayland-server.hpp>" : "#include <wayland-client.hpp>") << std::endl;
