--- afpfs-ng-0.8.1/include/afp.h	2011-08-25 20:58:17.000000000 +0200
+++ afpfs-ng-0.8.1/include/afp.h	2011-08-25 21:15:14.000000000 +0200
@@ -7,8 +7,8 @@
 #include <netdb.h>
 #include <sys/statvfs.h>
 #include <pwd.h>
-#include <afp_protocol.h>
-#include <libafpclient.h>
+#include <afp/afp_protocol.h>
+#include <afp/libafpclient.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
