--- configure.in-org	2011-11-08 11:53:42.802419355 -0500
+++ configure.in	2011-11-08 11:55:38.082419413 -0500
@@ -11,8 +11,9 @@ AC_PROG_LIBTOOL
 
 # Checks for libraries.
 #AC_CHECK_LIB([c],   [main])
-#AC_CHECK_LIB([m],   [main])
+AC_CHECK_LIB([m],   [main])
 AC_CHECK_LIB([ssl], [main],, AC_MSG_ERROR($missing_library))
+AC_CHECK_LIB([crypto], [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([pthread], [main],, AC_MSG_ERROR($missing_library))
 
 OUTPUT_FILES="Makefile"
@@ -21,4 +22,4 @@ LIBDIR=$PREFIX
 
 AC_CONFIG_FILES([${OUTPUT_FILES}])
 AC_OUTPUT(Makefile src/Makefile)
-AC_OUTPUT
\ No newline at end of file
+AC_OUTPUT

