--- src/socketlib.c	2012-01-04 20:41:05.000000000 +0100
+++ src/socketlib.c	2012-01-04 20:35:53.000000000 +0100
@@ -48,7 +48,7 @@
   if((tSock==-1) && (pAddrInfo->ai_family == AF_INET6) && (errno == EAFNOSUPPORT))
   {
     //Fallback to ipv4
-    perror("Failed to create ipv6 socket. Trying ipv4");
+    xprintf("Failed to create ipv6 socket. Trying ipv4");
     pAddrInfo->ai_family = AF_INET;
     tSock = socket(pAddrInfo->ai_family, pAddrInfo->ai_socktype, 0);
   }
@@ -158,8 +158,8 @@
     sprintf(tService, "%d", pPort); // copies port to string
     int tFamily = AF_INET;
     #ifdef AF_INET6
-    //printf("Listening on IPv6 Socket\n");
-    //tFamily = AF_INET6;
+    xprintf("Listening on IPv6 Socket\n");
+    tFamily = AF_INET6;
     #else
     //printf("Listening on IPv4 Socket");
     #endif
