diff -ur samba-3.0.28.org/source/client/mount.cifs.c samba-3.0.28/source/client/mount.cifs.c
--- samba-3.0.28.org/source/client/mount.cifs.c	2007-11-15 04:15:03.000000000 +0100
+++ samba-3.0.28/source/client/mount.cifs.c	2009-07-27 22:52:26.000000000 +0200
@@ -38,6 +38,7 @@
 #include <string.h>
 #include <mntent.h>
 #include <fcntl.h>
+#include <limits.h>
 
 #define MOUNT_CIFS_VERSION_MAJOR "1"
 #define MOUNT_CIFS_VERSION_MINOR "10"
diff -ur samba-3.0.28.org/source/configure samba-3.0.28/source/configure
--- samba-3.0.28.org/source/configure	2007-12-10 17:04:31.000000000 +0100
+++ samba-3.0.28/source/configure	2009-07-27 22:52:26.000000000 +0200
@@ -43617,7 +43617,7 @@
 
 
 
-for ac_header in linux/inotify.h asm/unistd.h
+for ac_header in linux/inotify.h asm/unistd.h sys/inotify.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
diff -ur samba-3.0.28.org/source/include/config.h.in samba-3.0.28/source/include/config.h.in
--- samba-3.0.28.org/source/include/config.h.in	2007-12-10 17:04:29.000000000 +0100
+++ samba-3.0.28/source/include/config.h.in	2009-07-27 22:52:26.000000000 +0200
@@ -1687,6 +1687,9 @@
 /* Define to 1 if you have the <sys/id.h> header file. */
 #undef HAVE_SYS_ID_H
 
+/* Define to 1 if you have the <sys/inotify.h> header file. */
+#undef HAVE_SYS_INOTIFY_H
+
 /* Define to 1 if you have the <sys/ioctl.h> header file. */
 #undef HAVE_SYS_IOCTL_H
 
diff -ur samba-3.0.28.org/source/smbd/notify_inotify.c samba-3.0.28/source/smbd/notify_inotify.c
--- samba-3.0.28.org/source/smbd/notify_inotify.c	2007-11-15 04:15:04.000000000 +0100
+++ samba-3.0.28/source/smbd/notify_inotify.c	2009-07-27 22:52:26.000000000 +0200
@@ -26,14 +26,20 @@
 
 #ifdef HAVE_INOTIFY
 
+#if HAVE_SYS_INOTIFY_H
+#include <sys/inotify.h>
+#else
+
 #ifdef HAVE_ASM_TYPES_H
 #include <asm/types.h>
 #endif
 
+
+#ifndef HAVE_INOTIFY_INIT
+
 #include <linux/inotify.h>
 #include <asm/unistd.h>
 
-#ifndef HAVE_INOTIFY_INIT
 /*
   glibc doesn't define these functions yet (as of March 2006)
 */
@@ -51,6 +57,11 @@
 {
 	return syscall(__NR_inotify_rm_watch, fd, wd);
 }
+#else
+
+#include <sys/inotify.h>
+
+#endif
 #endif
 
 
