--- libcedarv.orig/vdecoder.c	2012-09-21 16:48:15.821715289 +0200
+++ libcedarv/vdecoder.c	2012-09-21 16:48:39.549832952 +0200
@@ -690,9 +690,8 @@ get_frame_again:
     picture->pts                    = frame->pts;
     picture->pcr                    = frame->pcr;
 
-	//* the CedarX player need we return physic address of display frame.
-    picture->y                      = mem_get_phy_addr(frame->y);
-    picture->u                      = mem_get_phy_addr(frame->u);
+    picture->y                      = frame->y;
+    picture->u                      = frame->u;
 
     picture->v                      = NULL;
     picture->alpha                  = NULL;
