--- src/video/quartz/SDL_QuartzWindow.m.orig	2013-10-31 04:29:45.000000000 -0700
+++ src/video/quartz/SDL_QuartzWindow.m	2013-10-31 04:30:05.000000000 -0700
@@ -87,7 +87,7 @@
      SDL_VideoDevice *this = (SDL_VideoDevice*)current_video;

     /* make sure pixels are fully opaque */
-    if (! ( SDL_VideoSurface->flags & SDL_OPENGL ) )
+    if ( SDL_VideoSurface && ! ( SDL_VideoSurface->flags & SDL_OPENGL ) )
         QZ_SetPortAlphaOpaque ();

     /* save current visible SDL surface */
