--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,7 +318,7 @@
     ADD_DEFINITIONS(-DHAVE_OPENSSL -DHAVE_TLS)
     SET(SSL_SOURCES "${CC_SOURCE_DIR}/libmariadb/secure/openssl.c"
                     "${CC_SOURCE_DIR}/libmariadb/secure/openssl_crypt.c")
-    SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
+    SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY} ${_OPENSSL_LDFLAGS_OTHER})
     IF(WIN32 AND EXISTS ${OPENSSL_INCLUDE_DIR}/openssl/applink.c)
      SET(HAVE_OPENSSL_APPLINK_C 1)
     ENDIF()
