--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -497,9 +497,13 @@
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
 ENDIF()
 
+if(INSTALL_STATIC)
 INSTALL(TARGETS mariadbclient
           COMPONENT Development
           DESTINATION ${INSTALL_LIBDIR})
+endif()
+
+if(INSTALL_SHARED)
 IF(WIN32)
 INSTALL(TARGETS libmariadb
         COMPONENT SharedLibraries
@@ -514,6 +518,7 @@
 INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
         COMPONENT Development NAMELINK_ONLY)
 ENDIF()
+endif()
 
 IF(MSVC)
    # On Windows, install PDB
